<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Applied Population Genetics</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A textbook for the use of R in spatial genetic data analysis.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Applied Population Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="media/TitlePage_small.png" />
  <meta property="og:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="github-repo" content="dyerlab/AMG" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Applied Population Genetics" />
  
  <meta name="twitter:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="twitter:image" content="media/TitlePage_small.png" />

<meta name="author" content="Rodney J. Dyer">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="programming.html">
<link rel="next" href="reproducable-research.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-0.8/htmlwidgets.js"></script>
<link href="libs/leaflet-0.7.7/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-0.7.7/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<link href="libs/leaflet-label-0.2.2/leaflet.label.css" rel="stylesheet" />
<script src="libs/leaflet-label-0.2.2/leaflet.label.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4-compressed.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4leaflet.js"></script>
<script src="libs/leaflet-binding-1.1.0/leaflet.js"></script>
<script src="libs/leaflet-providers-1.0.27/leaflet-providers.js"></script>
<script src="libs/leaflet-providers-plugin-1.1.0/leaflet-providers-plugin.js"></script>
<script src="libs/d3-4.5.0/d3.min.js"></script>
<script src="libs/sankey-1/sankey.js"></script>
<script src="libs/sankeyNetwork-binding-0.3/sankeyNetwork.js"></script>
<script src="libs/dendroNetwork-binding-0.3/dendroNetwork.js"></script>
<script src="libs/simpleNetwork-binding-0.3/simpleNetwork.js"></script>
<script src="libs/forceNetwork-binding-0.3/forceNetwork.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="media/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Applied Population Genetics</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#logistics"><i class="fa fa-check"></i>Logistics</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#typography"><i class="fa fa-check"></i>Typography</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#interactive-content"><i class="fa fa-check"></i>Interactive Content</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#dedication"><i class="fa fa-check"></i>Dedication</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html"><i class="fa fa-check"></i><b>1</b> The R Ecosystem</a><ul>
<li class="chapter" data-level="1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#getting-r-configured"><i class="fa fa-check"></i><b>1.1</b> Getting R Configured</a><ul>
<li class="chapter" data-level="1.1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
<li class="chapter" data-level="1.1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#cran"><i class="fa fa-check"></i><b>1.1.2</b> CRAN</a></li>
<li class="chapter" data-level="1.1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#github"><i class="fa fa-check"></i><b>1.1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.1.4" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#bioconductor"><i class="fa fa-check"></i><b>1.1.4</b> Bioconductor</a></li>
<li class="chapter" data-level="1.1.5" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#troublesome-packages"><i class="fa fa-check"></i><b>1.1.5</b> Troublesome Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#libraries-used-in-text"><i class="fa fa-check"></i><b>1.2</b> Libraries Used in Text</a></li>
<li class="chapter" data-level="1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#the-rstudio-environment"><i class="fa fa-check"></i><b>1.3</b> The RStudio Environment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>2</b> Data Types</a><ul>
<li class="chapter" data-level="2.1" data-path="data-types.html"><a href="data-types.html#numeric-data-types"><i class="fa fa-check"></i><b>2.1</b> Numeric Data Types</a><ul>
<li class="chapter" data-level="2.1.1" data-path="data-types.html"><a href="data-types.html#coercion-to-numeric"><i class="fa fa-check"></i><b>2.1.1</b> Coercion to Numeric</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-types.html"><a href="data-types.html#characters"><i class="fa fa-check"></i><b>2.2</b> Characters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="data-types.html"><a href="data-types.html#concatenation-of-characters"><i class="fa fa-check"></i><b>2.2.1</b> Concatenation of Characters</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types.html"><a href="data-types.html#coercion-to-characters"><i class="fa fa-check"></i><b>2.2.2</b> Coercion to Characters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types.html"><a href="data-types.html#factors"><i class="fa fa-check"></i><b>2.3</b> Factors</a><ul>
<li class="chapter" data-level="2.3.1" data-path="data-types.html"><a href="data-types.html#missing-levels-in-factors"><i class="fa fa-check"></i><b>2.3.1</b> Missing Levels in Factors</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-types.html"><a href="data-types.html#logical-types"><i class="fa fa-check"></i><b>2.4</b> Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-containers.html"><a href="data-containers.html"><i class="fa fa-check"></i><b>3</b> Data Containers</a><ul>
<li class="chapter" data-level="3.1" data-path="data-containers.html"><a href="data-containers.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a></li>
<li class="chapter" data-level="3.2" data-path="data-containers.html"><a href="data-containers.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="data-containers.html"><a href="data-containers.html#lists"><i class="fa fa-check"></i><b>3.3</b> Lists</a></li>
<li class="chapter" data-level="3.4" data-path="data-containers.html"><a href="data-containers.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a><ul>
<li class="chapter" data-level="3.4.1" data-path="data-containers.html"><a href="data-containers.html#indexing-data-frames"><i class="fa fa-check"></i><b>3.4.1</b> Indexing Data Frames</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="programming.html"><a href="programming.html"><i class="fa fa-check"></i><b>4</b> Programming</a><ul>
<li class="chapter" data-level="4.1" data-path="programming.html"><a href="programming.html#function-writing"><i class="fa fa-check"></i><b>4.1</b> Function Writing</a></li>
<li class="chapter" data-level="4.2" data-path="programming.html"><a href="programming.html#variable-scope"><i class="fa fa-check"></i><b>4.2</b> Variable Scope</a></li>
<li class="chapter" data-level="4.3" data-path="programming.html"><a href="programming.html#decision-making"><i class="fa fa-check"></i><b>4.3</b> Decision Making</a><ul>
<li class="chapter" data-level="4.3.1" data-path="programming.html"><a href="programming.html#the-if-pattern"><i class="fa fa-check"></i><b>4.3.1</b> The if Pattern</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="programming.html"><a href="programming.html#the-if-else-pattern"><i class="fa fa-check"></i><b>4.4</b> The <code>if-else</code> Pattern</a></li>
<li class="chapter" data-level="4.5" data-path="programming.html"><a href="programming.html#flow-control"><i class="fa fa-check"></i><b>4.5</b> Flow Control</a><ul>
<li class="chapter" data-level="4.5.1" data-path="programming.html"><a href="programming.html#the-for-loop"><i class="fa fa-check"></i><b>4.5.1</b> The for() Loop</a></li>
<li class="chapter" data-level="4.5.2" data-path="programming.html"><a href="programming.html#short-circuiting-the-loop"><i class="fa fa-check"></i><b>4.5.2</b> Short Circuiting the Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>5</b> Graphics</a><ul>
<li class="chapter" data-level="5.1" data-path="graphics.html"><a href="graphics.html#univariate-data"><i class="fa fa-check"></i><b>5.1</b> Univariate Data</a></li>
<li class="chapter" data-level="5.2" data-path="graphics.html"><a href="graphics.html#plotting-parameters"><i class="fa fa-check"></i><b>5.2</b> Plotting Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="graphics.html"><a href="graphics.html#density-plots"><i class="fa fa-check"></i><b>5.3</b> Density Plots</a></li>
<li class="chapter" data-level="5.4" data-path="graphics.html"><a href="graphics.html#overlaying-plots"><i class="fa fa-check"></i><b>5.4</b> Overlaying Plots</a></li>
<li class="chapter" data-level="5.5" data-path="graphics.html"><a href="graphics.html#boxplots"><i class="fa fa-check"></i><b>5.5</b> Boxplots</a></li>
<li class="chapter" data-level="5.6" data-path="graphics.html"><a href="graphics.html#scatter-plots"><i class="fa fa-check"></i><b>5.6</b> Scatter Plots</a></li>
<li class="chapter" data-level="5.7" data-path="graphics.html"><a href="graphics.html#multiple-plots"><i class="fa fa-check"></i><b>5.7</b> Multiple Plots</a></li>
<li class="chapter" data-level="5.8" data-path="graphics.html"><a href="graphics.html#color-palettes"><i class="fa fa-check"></i><b>5.8</b> Color Palettes</a></li>
<li class="chapter" data-level="5.9" data-path="graphics.html"><a href="graphics.html#rcolorbrewer"><i class="fa fa-check"></i><b>5.9</b> RColorBrewer</a></li>
<li class="chapter" data-level="5.10" data-path="graphics.html"><a href="graphics.html#saving-imagery"><i class="fa fa-check"></i><b>5.10</b> Saving Imagery</a></li>
<li class="chapter" data-level="5.11" data-path="graphics.html"><a href="graphics.html#interactive-graphics"><i class="fa fa-check"></i><b>5.11</b> Interactive Graphics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="reproducable-research.html"><a href="reproducable-research.html"><i class="fa fa-check"></i><b>6</b> Reproducable Research</a><ul>
<li class="chapter" data-level="6.1" data-path="reproducable-research.html"><a href="reproducable-research.html#keeping-data-on-google-drive"><i class="fa fa-check"></i><b>6.1</b> Keeping Data on Google Drive</a></li>
<li class="chapter" data-level="6.2" data-path="reproducable-research.html"><a href="reproducable-research.html#integrating-into-slack"><i class="fa fa-check"></i><b>6.2</b> Integrating into Slack</a><ul>
<li class="chapter" data-level="6.2.1" data-path="reproducable-research.html"><a href="reproducable-research.html#fixing-problem-with-version-1.4.1"><i class="fa fa-check"></i><b>6.2.1</b> Fixing Problem with Version 1.4.1</a></li>
<li class="chapter" data-level="6.2.2" data-path="reproducable-research.html"><a href="reproducable-research.html#configuration"><i class="fa fa-check"></i><b>6.2.2</b> Configuration</a></li>
<li class="chapter" data-level="6.2.3" data-path="reproducable-research.html"><a href="reproducable-research.html#sending-text-and-graphical-ouput"><i class="fa fa-check"></i><b>6.2.3</b> Sending Text and Graphical Ouput</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>I Single Population</b></span></li>
<li class="chapter" data-level="7" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html"><i class="fa fa-check"></i><b>7</b> Hardy-Weinberg Equilibrium</a><ul>
<li class="chapter" data-level="7.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#genetic-assumptions"><i class="fa fa-check"></i><b>7.1</b> Genetic Assumptions</a><ul>
<li class="chapter" data-level="7.1.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#punnet-squares"><i class="fa fa-check"></i><b>7.1.1</b> Punnet Squares</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#evolutionary-assumptions"><i class="fa fa-check"></i><b>7.2</b> Evolutionary Assumptions</a></li>
<li class="chapter" data-level="7.3" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#the-mechanics"><i class="fa fa-check"></i><b>7.3</b> The Mechanics</a></li>
<li class="chapter" data-level="7.4" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#trans-generational-transition-probabilities"><i class="fa fa-check"></i><b>7.4</b> Trans-Generational Transition Probabilities</a></li>
<li class="chapter" data-level="7.5" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#consequences-of-hwe"><i class="fa fa-check"></i><b>7.5</b> Consequences of HWE</a></li>
<li class="chapter" data-level="7.6" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#zygotic-gene-flow"><i class="fa fa-check"></i><b>7.6</b> Zygotic Gene Flow</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="genetic-drift.html"><a href="genetic-drift.html"><i class="fa fa-check"></i><b>8</b> Genetic Drift</a><ul>
<li class="chapter" data-level="8.1" data-path="genetic-drift.html"><a href="genetic-drift.html#consequences-of-genetic-drift"><i class="fa fa-check"></i><b>8.1</b> Consequences of Genetic Drift</a><ul>
<li class="chapter" data-level="8.1.1" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-fixation"><i class="fa fa-check"></i><b>8.1.1</b> Time to Fixation</a></li>
<li class="chapter" data-level="8.1.2" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-allele-loss"><i class="fa fa-check"></i><b>8.1.2</b> Time to Allele Loss</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="genetic-drift.html"><a href="genetic-drift.html#mutation-drift"><i class="fa fa-check"></i><b>8.2</b> Mutation &amp; Drift</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="effective-population-size.html"><a href="effective-population-size.html"><i class="fa fa-check"></i><b>9</b> Effective Population Size</a><ul>
<li class="chapter" data-level="9.1" data-path="effective-population-size.html"><a href="effective-population-size.html#demographic-effective-population-sizes"><i class="fa fa-check"></i><b>9.1</b> Demographic Effective Population Sizes</a></li>
<li class="chapter" data-level="9.2" data-path="effective-population-size.html"><a href="effective-population-size.html#genetic-effective-population-size"><i class="fa fa-check"></i><b>9.2</b> Genetic Effective Population Size</a><ul>
<li class="chapter" data-level="9.2.1" data-path="effective-population-size.html"><a href="effective-population-size.html#inbreeding-effective-population-size"><i class="fa fa-check"></i><b>9.2.1</b> Inbreeding Effective Population Size</a></li>
<li class="chapter" data-level="9.2.2" data-path="effective-population-size.html"><a href="effective-population-size.html#variance-effective-population-size"><i class="fa fa-check"></i><b>9.2.2</b> Variance Effective Population Size</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="effective-population-size.html"><a href="effective-population-size.html#fluctuating-population-size"><i class="fa fa-check"></i><b>9.3</b> Fluctuating Population Size</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutation.html"><a href="mutation.html"><i class="fa fa-check"></i><b>10</b> Mutation</a><ul>
<li class="chapter" data-level="10.1" data-path="mutation.html"><a href="mutation.html#mutation-models"><i class="fa fa-check"></i><b>10.1</b> Mutation Models</a><ul>
<li class="chapter" data-level="10.1.1" data-path="mutation.html"><a href="mutation.html#fixed-allele-model"><i class="fa fa-check"></i><b>10.1.1</b> Fixed Allele Model</a></li>
<li class="chapter" data-level="10.1.2" data-path="mutation.html"><a href="mutation.html#infinite-allele-model"><i class="fa fa-check"></i><b>10.1.2</b> Infinite Allele Model</a></li>
<li class="chapter" data-level="10.1.3" data-path="mutation.html"><a href="mutation.html#stepwise-mutation-model"><i class="fa fa-check"></i><b>10.1.3</b> Stepwise Mutation Model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="mutation.html"><a href="mutation.html#mutation-and-inbreeding"><i class="fa fa-check"></i><b>10.2</b> Mutation and Inbreeding</a></li>
<li class="chapter" data-level="10.3" data-path="mutation.html"><a href="mutation.html#estimating-mutation"><i class="fa fa-check"></i><b>10.3</b> Estimating Mutation</a><ul>
<li class="chapter" data-level="10.3.1" data-path="mutation.html"><a href="mutation.html#phenotypes"><i class="fa fa-check"></i><b>10.3.1</b> Phenotypes</a></li>
<li class="chapter" data-level="10.3.2" data-path="mutation.html"><a href="mutation.html#genotypes"><i class="fa fa-check"></i><b>10.3.2</b> Genotypes</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Nonrandom Rating</b></span></li>
<li class="chapter" data-level="11" data-path="selfing.html"><a href="selfing.html"><i class="fa fa-check"></i><b>11</b> Selfing</a><ul>
<li class="chapter" data-level="11.1" data-path="selfing.html"><a href="selfing.html#changes-in-f"><i class="fa fa-check"></i><b>11.1</b> Changes in F</a><ul>
<li class="chapter" data-level="11.1.1" data-path="selfing.html"><a href="selfing.html#theoretical-expectations"><i class="fa fa-check"></i><b>11.1.1</b> Theoretical Expectations</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="selfing.html"><a href="selfing.html#simulation-example"><i class="fa fa-check"></i><b>11.2</b> Simulation Example</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html"><i class="fa fa-check"></i><b>12</b> Mixed Mating Systems</a><ul>
<li class="chapter" data-level="12.1" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#mixed-mating-genotype-frequency-equilibriums"><i class="fa fa-check"></i><b>12.1</b> Mixed Mating Genotype Frequency Equilibriums</a></li>
<li class="chapter" data-level="12.2" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#estimating-selfing-rates"><i class="fa fa-check"></i><b>12.2</b> Estimating Selfing Rates</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wahlund-effects.html"><a href="wahlund-effects.html"><i class="fa fa-check"></i><b>13</b> Wahlund Effects</a></li>
<li class="part"><span><b>III Parent Offspring Data</b></span></li>
<li class="chapter" data-level="14" data-path="pedigrees.html"><a href="pedigrees.html"><i class="fa fa-check"></i><b>14</b> Pedigrees</a><ul>
<li class="chapter" data-level="14.1" data-path="pedigrees.html"><a href="pedigrees.html#expectations-for-f-in-a-pedigree."><i class="fa fa-check"></i><b>14.1</b> Expectations for F in a pedigree.</a></li>
<li class="chapter" data-level="14.2" data-path="pedigrees.html"><a href="pedigrees.html#drawing-pedigrees"><i class="fa fa-check"></i><b>14.2</b> Drawing Pedigrees</a></li>
<li class="chapter" data-level="14.3" data-path="pedigrees.html"><a href="pedigrees.html#skills"><i class="fa fa-check"></i><b>14.3</b> Skills</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="parentage-analyses.html"><a href="parentage-analyses.html"><i class="fa fa-check"></i><b>15</b> Parentage Analyses</a><ul>
<li class="chapter" data-level="15.1" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-vs.maternity"><i class="fa fa-check"></i><b>15.1</b> Paternity vs. Maternity</a></li>
<li class="chapter" data-level="15.2" data-path="parentage-analyses.html"><a href="parentage-analyses.html#probability-of-exclusion"><i class="fa fa-check"></i><b>15.2</b> Probability of Exclusion</a></li>
<li class="chapter" data-level="15.3" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-exclusion"><i class="fa fa-check"></i><b>15.3</b> Paternity Exclusion</a></li>
<li class="chapter" data-level="15.4" data-path="parentage-analyses.html"><a href="parentage-analyses.html#fractional-paternity"><i class="fa fa-check"></i><b>15.4</b> Fractional Paternity</a></li>
<li class="chapter" data-level="15.5" data-path="parentage-analyses.html"><a href="parentage-analyses.html#dispersal-kernels"><i class="fa fa-check"></i><b>15.5</b> Dispersal Kernels</a></li>
</ul></li>
<li class="part"><span><b>IV Subdivided Populations</b></span></li>
<li class="chapter" data-level="16" data-path="mapping-populations.html"><a href="mapping-populations.html"><i class="fa fa-check"></i><b>16</b> Mapping Populations</a><ul>
<li class="chapter" data-level="16.1" data-path="mapping-populations.html"><a href="mapping-populations.html#procedure"><i class="fa fa-check"></i><b>16.1</b> Procedure</a></li>
<li class="chapter" data-level="16.2" data-path="mapping-populations.html"><a href="mapping-populations.html#plotting-population-locations"><i class="fa fa-check"></i><b>16.2</b> Plotting Population Locations</a></li>
<li class="chapter" data-level="16.3" data-path="mapping-populations.html"><a href="mapping-populations.html#population-allele-frequencies"><i class="fa fa-check"></i><b>16.3</b> Population Allele Frequencies</a></li>
<li class="chapter" data-level="16.4" data-path="mapping-populations.html"><a href="mapping-populations.html#dynamic-maps"><i class="fa fa-check"></i><b>16.4</b> Dynamic maps</a><ul>
<li class="chapter" data-level="16.4.1" data-path="mapping-populations.html"><a href="mapping-populations.html#sankey-networks"><i class="fa fa-check"></i><b>16.4.1</b> Sankey Networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="population-models.html"><a href="population-models.html"><i class="fa fa-check"></i><b>17</b> Population Models</a><ul>
<li class="chapter" data-level="17.1" data-path="population-models.html"><a href="population-models.html#models-of-partition"><i class="fa fa-check"></i><b>17.1</b> Models of Partition</a></li>
<li class="chapter" data-level="17.2" data-path="population-models.html"><a href="population-models.html#island-mainland-model"><i class="fa fa-check"></i><b>17.2</b> Island Mainland Model</a></li>
<li class="chapter" data-level="17.3" data-path="population-models.html"><a href="population-models.html#the-island-model"><i class="fa fa-check"></i><b>17.3</b> The Island Model</a></li>
<li class="chapter" data-level="17.4" data-path="population-models.html"><a href="population-models.html#stepping-stone-models"><i class="fa fa-check"></i><b>17.4</b> Stepping Stone Models</a></li>
<li class="chapter" data-level="17.5" data-path="population-models.html"><a href="population-models.html#the-general-model"><i class="fa fa-check"></i><b>17.5</b> The General Model</a></li>
<li class="chapter" data-level="17.6" data-path="population-models.html"><a href="population-models.html#general-strategies"><i class="fa fa-check"></i><b>17.6</b> General Strategies</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>18</b> Genetic Diversity</a><ul>
<li class="chapter" data-level="18.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#allelic-diversity"><i class="fa fa-check"></i><b>18.1</b> Allelic Diversity</a></li>
<li class="chapter" data-level="18.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#genotypic-diversity"><i class="fa fa-check"></i><b>18.2</b> Genotypic Diversity</a></li>
<li class="chapter" data-level="18.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#samples-from-several-locales"><i class="fa fa-check"></i><b>18.3</b> Samples from Several Locales</a></li>
<li class="chapter" data-level="18.4" data-path="genetic-diversity.html"><a href="genetic-diversity.html#multilocus-diversity"><i class="fa fa-check"></i><b>18.4</b> Multilocus Diversity</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="rarefaction.html"><a href="rarefaction.html"><i class="fa fa-check"></i><b>19</b> Rarefaction</a><ul>
<li class="chapter" data-level="19.1" data-path="rarefaction.html"><a href="rarefaction.html#mapping-diversity"><i class="fa fa-check"></i><b>19.1</b> Mapping Diversity</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="genetic-distances.html"><a href="genetic-distances.html"><i class="fa fa-check"></i><b>20</b> Genetic Distances</a><ul>
<li class="chapter" data-level="20.1" data-path="genetic-distances.html"><a href="genetic-distances.html#individual-genetic-distance"><i class="fa fa-check"></i><b>20.1</b> Individual Genetic Distance</a><ul>
<li class="chapter" data-level="20.1.1" data-path="genetic-distances.html"><a href="genetic-distances.html#euclidean-distance"><i class="fa fa-check"></i><b>20.1.1</b> Euclidean Distance</a></li>
<li class="chapter" data-level="20.1.2" data-path="genetic-distances.html"><a href="genetic-distances.html#amova-distance"><i class="fa fa-check"></i><b>20.1.2</b> AMOVA Distance</a></li>
<li class="chapter" data-level="20.1.3" data-path="genetic-distances.html"><a href="genetic-distances.html#bray-curtis-individual-distance"><i class="fa fa-check"></i><b>20.1.3</b> Bray-Curtis Individual Distance</a></li>
<li class="chapter" data-level="20.1.4" data-path="genetic-distances.html"><a href="genetic-distances.html#ladder-distance"><i class="fa fa-check"></i><b>20.1.4</b> Ladder Distance</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="genetic-distances.html"><a href="genetic-distances.html#population-level-genetic-distances"><i class="fa fa-check"></i><b>20.2</b> Population-Level Genetic Distances</a><ul>
<li class="chapter" data-level="20.2.1" data-path="genetic-distances.html"><a href="genetic-distances.html#czekanowski-manhattan-distance"><i class="fa fa-check"></i><b>20.2.1</b> Czekanowski (Manhattan) Distance</a></li>
<li class="chapter" data-level="20.2.2" data-path="genetic-distances.html"><a href="genetic-distances.html#rogers-euclidean-distance"><i class="fa fa-check"></i><b>20.2.2</b> Rogers (Euclidean) Distance</a></li>
<li class="chapter" data-level="20.2.3" data-path="genetic-distances.html"><a href="genetic-distances.html#neis-genetic-distance"><i class="fa fa-check"></i><b>20.2.3</b> Nei’s Genetic Distance</a></li>
<li class="chapter" data-level="20.2.4" data-path="genetic-distances.html"><a href="genetic-distances.html#conditional-genetic-distance"><i class="fa fa-check"></i><b>20.2.4</b> Conditional Genetic Distance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="genetic-structure.html"><a href="genetic-structure.html"><i class="fa fa-check"></i><b>21</b> Genetic Structure</a><ul>
<li class="chapter" data-level="21.1" data-path="genetic-structure.html"><a href="genetic-structure.html#f-statistics-and-thier-ilk"><i class="fa fa-check"></i><b>21.1</b> F-Statistics and Thier Ilk</a><ul>
<li class="chapter" data-level="21.1.1" data-path="genetic-structure.html"><a href="genetic-structure.html#issues-with-fixation-indices"><i class="fa fa-check"></i><b>21.1.1</b> Issues with Fixation Indices</a></li>
<li class="chapter" data-level="21.1.2" data-path="genetic-structure.html"><a href="genetic-structure.html#additional-x_st-like-parameters"><i class="fa fa-check"></i><b>21.1.2</b> Additional <span class="math inline">\(X_{ST}\)</span>-like Parameters</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="genetic-structure.html"><a href="genetic-structure.html#statistical-structure"><i class="fa fa-check"></i><b>21.2</b> Statistical Structure</a></li>
<li class="chapter" data-level="21.3" data-path="genetic-structure.html"><a href="genetic-structure.html#final-thoughts-on-structure"><i class="fa fa-check"></i><b>21.3</b> Final Thoughts on Structure</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="population-assignment.html"><a href="population-assignment.html"><i class="fa fa-check"></i><b>22</b> Population Assignment</a></li>
<li class="chapter" data-level="23" data-path="admixture.html"><a href="admixture.html"><i class="fa fa-check"></i><b>23</b> Admixture</a></li>
<li class="part"><span><b>V Statistical Models</b></span></li>
<li class="chapter" data-level="24" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html"><i class="fa fa-check"></i><b>24</b> Ordination of Genetic Data</a><ul>
<li class="chapter" data-level="24.1" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#hierarchical-clustering"><i class="fa fa-check"></i><b>24.1</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="24.2" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#interactive-tree-display"><i class="fa fa-check"></i><b>24.2</b> Interactive Tree Display</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="distance-matrices.html"><a href="distance-matrices.html"><i class="fa fa-check"></i><b>25</b> Distance Matrices</a><ul>
<li class="chapter" data-level="25.1" data-path="distance-matrices.html"><a href="distance-matrices.html#visual-comparisons"><i class="fa fa-check"></i><b>25.1</b> Visual Comparisons</a></li>
<li class="chapter" data-level="25.2" data-path="distance-matrices.html"><a href="distance-matrices.html#the-mantel-test"><i class="fa fa-check"></i><b>25.2</b> The Mantel Test</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="linear-models.html"><a href="linear-models.html"><i class="fa fa-check"></i><b>26</b> Linear Models</a><ul>
<li class="chapter" data-level="26.1" data-path="linear-models.html"><a href="linear-models.html#dependent-independent-relationships"><i class="fa fa-check"></i><b>26.1</b> Dependent &amp; Independent Relationships</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="distance-networks.html"><a href="distance-networks.html"><i class="fa fa-check"></i><b>27</b> Distance Networks</a><ul>
<li class="chapter" data-level="27.1" data-path="distance-networks.html"><a href="distance-networks.html#graph-structure"><i class="fa fa-check"></i><b>27.1</b> Graph Structure</a><ul>
<li class="chapter" data-level="27.1.1" data-path="distance-networks.html"><a href="distance-networks.html#node-edge-attributes"><i class="fa fa-check"></i><b>27.1.1</b> Node &amp; Edge Attributes</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="distance-networks.html"><a href="distance-networks.html#plotting-a-graphs"><i class="fa fa-check"></i><b>27.2</b> Plotting a Graphs</a></li>
<li class="chapter" data-level="27.3" data-path="distance-networks.html"><a href="distance-networks.html#mapping-networks"><i class="fa fa-check"></i><b>27.3</b> Mapping Networks</a></li>
<li class="chapter" data-level="27.4" data-path="distance-networks.html"><a href="distance-networks.html#genetic-distance-graphs"><i class="fa fa-check"></i><b>27.4</b> Genetic Distance Graphs</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="population-graphs.html"><a href="population-graphs.html"><i class="fa fa-check"></i><b>28</b> Population Graphs</a><ul>
<li class="chapter" data-level="28.1" data-path="population-graphs.html"><a href="population-graphs.html#adding-data-to-a-graph"><i class="fa fa-check"></i><b>28.1</b> Adding data to a graph</a><ul>
<li class="chapter" data-level="28.1.1" data-path="population-graphs.html"><a href="population-graphs.html#plotting-a-graph-using-ggplot2-routines"><i class="fa fa-check"></i><b>28.1.1</b> Plotting a graph using <strong>ggplot2</strong> routines</a></li>
<li class="chapter" data-level="28.1.2" data-path="population-graphs.html"><a href="population-graphs.html#reading-existing-popgraph-files"><i class="fa fa-check"></i><b>28.1.2</b> Reading Existing <strong>popgraph</strong> Files</a></li>
<li class="chapter" data-level="28.1.3" data-path="population-graphs.html"><a href="population-graphs.html#saving-population-graph-objects"><i class="fa fa-check"></i><b>28.1.3</b> Saving Population Graph Objects</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="population-graphs.html"><a href="population-graphs.html#interactive-networks"><i class="fa fa-check"></i><b>28.2</b> Interactive Networks</a></li>
<li class="chapter" data-level="28.3" data-path="population-graphs.html"><a href="population-graphs.html#spatial-population-graphs"><i class="fa fa-check"></i><b>28.3</b> Spatial Population Graphs</a><ul>
<li class="chapter" data-level="28.3.1" data-path="population-graphs.html"><a href="population-graphs.html#integrating-google-and-ggplot2-for-plotting"><i class="fa fa-check"></i><b>28.3.1</b> Integrating Google and <strong>ggplot2</strong> for Plotting</a></li>
<li class="chapter" data-level="28.3.2" data-path="population-graphs.html"><a href="population-graphs.html#integrating-raster-maps"><i class="fa fa-check"></i><b>28.3.2</b> Integrating Raster Maps</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="population-graphs.html"><a href="population-graphs.html#extracting-spatial-data-using-population-graphs"><i class="fa fa-check"></i><b>28.4</b> Extracting Spatial Data Using Population Graphs</a><ul>
<li class="chapter" data-level="28.4.1" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-data"><i class="fa fa-check"></i><b>28.4.1</b> Node Specific Data</a></li>
<li class="chapter" data-level="28.4.2" data-path="population-graphs.html"><a href="population-graphs.html#extracting-data-along-popgraph-edges"><i class="fa fa-check"></i><b>28.4.2</b> Extracting Data Along Popgraph Edges</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="population-graphs.html"><a href="population-graphs.html#extracting-graph-theoretic-parameters"><i class="fa fa-check"></i><b>28.5</b> Extracting Graph-Theoretic Parameters</a><ul>
<li class="chapter" data-level="28.5.1" data-path="population-graphs.html"><a href="population-graphs.html#matrix-representations-of-population-graph-objects"><i class="fa fa-check"></i><b>28.5.1</b> Matrix Representations of Population Graph Objects</a></li>
<li class="chapter" data-level="28.5.2" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-parameters"><i class="fa fa-check"></i><b>28.5.2</b> Node Specific Parameters</a></li>
<li class="chapter" data-level="28.5.3" data-path="population-graphs.html"><a href="population-graphs.html#edge-specific-parameters"><i class="fa fa-check"></i><b>28.5.3</b> Edge Specific Parameters</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="population-graphs.html"><a href="population-graphs.html#testing-for-topological-congruence"><i class="fa fa-check"></i><b>28.6</b> Testing for Topological Congruence</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="interactive-visualization.html"><a href="interactive-visualization.html"><i class="fa fa-check"></i><b>29</b> Interactive Visualization</a><ul>
<li class="chapter" data-level="29.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#procedure-1"><i class="fa fa-check"></i><b>29.1</b> Procedure</a><ul>
<li class="chapter" data-level="29.1.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#interactive-networks-1"><i class="fa fa-check"></i><b>29.1.1</b> Interactive Networks</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>VI Spatial Data</b></span></li>
<li class="chapter" data-level="30" data-path="vector-data.html"><a href="vector-data.html"><i class="fa fa-check"></i><b>30</b> Vector Data</a><ul>
<li class="chapter" data-level="30.1" data-path="vector-data.html"><a href="vector-data.html#points"><i class="fa fa-check"></i><b>30.1</b> Points</a></li>
<li class="chapter" data-level="30.2" data-path="vector-data.html"><a href="vector-data.html#lines"><i class="fa fa-check"></i><b>30.2</b> Lines</a></li>
<li class="chapter" data-level="30.3" data-path="vector-data.html"><a href="vector-data.html#polygons"><i class="fa fa-check"></i><b>30.3</b> Polygons</a></li>
<li class="chapter" data-level="30.4" data-path="vector-data.html"><a href="vector-data.html#saving-vector-objects"><i class="fa fa-check"></i><b>30.4</b> Saving Vector Objects</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="raster-data.html"><a href="raster-data.html"><i class="fa fa-check"></i><b>31</b> Raster Data</a><ul>
<li class="chapter" data-level="31.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters"><i class="fa fa-check"></i><b>31.1</b> Cropping Rasters</a><ul>
<li class="chapter" data-level="31.1.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-polygons"><i class="fa fa-check"></i><b>31.1.1</b> Cropping Rasters Via Polygons</a></li>
<li class="chapter" data-level="31.1.2" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-convex-hull"><i class="fa fa-check"></i><b>31.1.2</b> Cropping Rasters Via Convex Hull</a></li>
</ul></li>
<li class="chapter" data-level="31.2" data-path="raster-data.html"><a href="raster-data.html#modifying-rasters"><i class="fa fa-check"></i><b>31.2</b> Modifying Rasters</a></li>
<li class="chapter" data-level="31.3" data-path="raster-data.html"><a href="raster-data.html#extracting-point-data-from-rasters"><i class="fa fa-check"></i><b>31.3</b> Extracting Point Data From Rasters</a><ul>
<li class="chapter" data-level="31.3.1" data-path="raster-data.html"><a href="raster-data.html#stacks-of-rasters"><i class="fa fa-check"></i><b>31.3.1</b> Stacks of Rasters</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="raster-data.html"><a href="raster-data.html#rasters-ggplot"><i class="fa fa-check"></i><b>31.4</b> Rasters &amp; ggplot</a></li>
<li class="chapter" data-level="31.5" data-path="raster-data.html"><a href="raster-data.html#d-visualization"><i class="fa fa-check"></i><b>31.5</b> 3D Visualization</a></li>
<li class="chapter" data-level="31.6" data-path="raster-data.html"><a href="raster-data.html#saving-exporting-rasters"><i class="fa fa-check"></i><b>31.6</b> Saving &amp; Exporting Rasters</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="map-projections.html"><a href="map-projections.html"><i class="fa fa-check"></i><b>32</b> Map Projections</a><ul>
<li class="chapter" data-level="32.1" data-path="map-projections.html"><a href="map-projections.html#projections"><i class="fa fa-check"></i><b>32.1</b> Projections</a><ul>
<li class="chapter" data-level="32.1.1" data-path="map-projections.html"><a href="map-projections.html#equatorial-projections"><i class="fa fa-check"></i><b>32.1.1</b> Equatorial Projections</a></li>
<li class="chapter" data-level="32.1.2" data-path="map-projections.html"><a href="map-projections.html#azimuth-projections"><i class="fa fa-check"></i><b>32.1.2</b> Azimuth Projections</a></li>
<li class="chapter" data-level="32.1.3" data-path="map-projections.html"><a href="map-projections.html#polar-conic-projections"><i class="fa fa-check"></i><b>32.1.3</b> Polar Conic Projections</a></li>
<li class="chapter" data-level="32.1.4" data-path="map-projections.html"><a href="map-projections.html#miscellaneous-projections"><i class="fa fa-check"></i><b>32.1.4</b> Miscellaneous Projections</a></li>
<li class="chapter" data-level="32.1.5" data-path="map-projections.html"><a href="map-projections.html#reprojecting-rasters"><i class="fa fa-check"></i><b>32.1.5</b> Reprojecting Rasters</a></li>
<li class="chapter" data-level="32.1.6" data-path="map-projections.html"><a href="map-projections.html#projecting-in-ggplot"><i class="fa fa-check"></i><b>32.1.6</b> Projecting In GGPlot</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="map-projections.html"><a href="map-projections.html#coordinate-systems"><i class="fa fa-check"></i><b>32.2</b> Coordinate Systems</a><ul>
<li class="chapter" data-level="32.2.1" data-path="map-projections.html"><a href="map-projections.html#changing-datum"><i class="fa fa-check"></i><b>32.2.1</b> Changing Datum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="ecological-distance.html"><a href="ecological-distance.html"><i class="fa fa-check"></i><b>33</b> Ecological Distance</a><ul>
<li class="chapter" data-level="33.1" data-path="ecological-distance.html"><a href="ecological-distance.html#the-concept-of-resistance"><i class="fa fa-check"></i><b>33.1</b> The Concept of Resistance</a><ul>
<li class="chapter" data-level="33.1.1" data-path="ecological-distance.html"><a href="ecological-distance.html#absolute-resistance"><i class="fa fa-check"></i><b>33.1.1</b> Absolute Resistance</a></li>
</ul></li>
<li class="chapter" data-level="33.2" data-path="ecological-distance.html"><a href="ecological-distance.html#relative-resistance"><i class="fa fa-check"></i><b>33.2</b> Relative Resistance</a></li>
<li class="chapter" data-level="33.3" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-separation"><i class="fa fa-check"></i><b>33.3</b> Estimating Separation</a><ul>
<li class="chapter" data-level="33.3.1" data-path="ecological-distance.html"><a href="ecological-distance.html#least-cost-distance"><i class="fa fa-check"></i><b>33.3.1</b> Least Cost Distance</a></li>
<li class="chapter" data-level="33.3.2" data-path="ecological-distance.html"><a href="ecological-distance.html#all-paths-circuit-distance"><i class="fa fa-check"></i><b>33.3.2</b> All Paths (Circuit) Distance</a></li>
<li class="chapter" data-level="33.3.3" data-path="ecological-distance.html"><a href="ecological-distance.html#path-corridor-constraints"><i class="fa fa-check"></i><b>33.3.3</b> Path &amp; Corridor Constraints</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-correlations-in-distance"><i class="fa fa-check"></i><b>33.4</b> Estimating Correlations in Distance</a><ul>
<li class="chapter" data-level="33.4.1" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-least-cost-path"><i class="fa fa-check"></i><b>33.4.1</b> Pairwise Least Cost Path</a></li>
<li class="chapter" data-level="33.4.2" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-all-pairs"><i class="fa fa-check"></i><b>33.4.2</b> Pairwise All Pairs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="niche-modeling.html"><a href="niche-modeling.html"><i class="fa fa-check"></i><b>34</b> Niche Modeling</a></li>
<li class="chapter" data-level="35" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html"><i class="fa fa-check"></i><b>35</b> Eigenvector Maps</a></li>
<li class="part"><span><b>VII Selection</b></span></li>
<li class="chapter" data-level="36" data-path="classical-selection-models.html"><a href="classical-selection-models.html"><i class="fa fa-check"></i><b>36</b> Classical Selection Models</a><ul>
<li class="chapter" data-level="36.1" data-path="classical-selection-models.html"><a href="classical-selection-models.html#the-strength-of-selection"><i class="fa fa-check"></i><b>36.1</b> The Strength of Selection</a></li>
</ul></li>
<li class="part"><span><b>VIII Backmatter</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="37" data-path="included-data-sets.html"><a href="included-data-sets.html"><i class="fa fa-check"></i><b>37</b> Included Data Sets</a><ul>
<li class="chapter" data-level="37.1" data-path="included-data-sets.html"><a href="included-data-sets.html#the-sonoran-desert-bark-beetle-araptus-attenuata"><i class="fa fa-check"></i><b>37.1</b> The Sonoran desert bark beetle, <em>Araptus attenuata</em></a><ul>
<li class="chapter" data-level="37.1.1" data-path="included-data-sets.html"><a href="included-data-sets.html#nuclear-markers"><i class="fa fa-check"></i><b>37.1.1</b> Nuclear Markers</a></li>
<li class="chapter" data-level="37.1.2" data-path="included-data-sets.html"><a href="included-data-sets.html#methylation-markers"><i class="fa fa-check"></i><b>37.1.2</b> Methylation Markers</a></li>
</ul></li>
<li class="chapter" data-level="37.2" data-path="included-data-sets.html"><a href="included-data-sets.html#the-flowering-dogwood-cornus-florida"><i class="fa fa-check"></i><b>37.2</b> The Flowering Dogwood <em>Cornus florida</em></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-author.html"><a href="about-the-author.html"><i class="fa fa-check"></i>About the Author</a></li>
<li class="divider"></li>
<li><a href="http://dyerlab.bio.vcu.edu" target="blank">the Dyer Laboratory</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Population Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="graphics" class="section level1 imageChapter">
<h1><span class="header-section-number">5</span> Graphics</h1>
<div class="chapter_image">
<img src="chapter_images/ch_fallleaves.JPG">
</div>
<p>One of the most critical features of data analysis is the ability to present your results in a logical and meaningful fashion. R has built-in functions that can provide you graphical output that will suffice for your understanding and interpretations. However, there are also third-party packages that make some truly beautiful output. In this section, both built-in graphics and graphical output from the ggplot2 library are explained and highlighted.</p>
<p>For this section we are going to use the venerable iris dataset that was used by Anderson (1935) and Fisher (1936). These data are measurements of four morphological variables (Sepal Length, Sepal Width, Petal Length, and Petal Width) measured on fifty individual iris plants from three recognized species. Here is a summary of this data set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(iris)</code></pre></div>
<pre><code>##   Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   
##  Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  
##  1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  
##  Median :5.800   Median :3.000   Median :4.350   Median :1.300  
##  Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  
##  3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  
##  Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  
##        Species  
##  setosa    :50  
##  versicolor:50  
##  virginica :50  
##                 
##                 
## </code></pre>
<p>I will also provide examples using two different plotting approaches. R has a robust set of built-in graphical routines that you can use. However, the development community has also provided several additional graphics libraries available for creating output. The one I prefer is <code>ggplot2</code> written by Hadley Wickham. His approach in designing this library a philosophy of graphics depicted in Leland Wilkson’s (2005) book <em>The Grammar of Graphics</em>. As I understand it, the idea is that a graphical display consists of several layers of information. These layers may include:</p>
<ul>
<li>The underlying data.<br />
</li>
<li>Mapping of the data onto one or more axes.<br />
</li>
<li>Geometric representations of data as points, lines, and/or areas.<br />
</li>
<li>Transformations of the axes into different coordinate spaces (e.g., cartesian, polar, etc.) or the data onto different scales (e.g., logrithmic)<br />
</li>
<li>Specification of subplots.</li>
</ul>
<p>In the normal plotting routines discussed before, configuration of these layers were specified as arguments passed to the plotting function (<code>plot()</code>, <code>boxplot()</code>, etc.). The ggplot2 library takes a different approach, allowing you to specify these components separately and literally add them together like components of a linear model.</p>
<div id="univariate-data" class="section level2">
<h2><span class="header-section-number">5.1</span> Univariate Data</h2>
<p>Univariate data can represent either counts (e.g., integers) of items or vectors of data that have decimal components (e.g., frequency distributions, etc.). If the sampling units are discrete, then a the <code>barplot()</code> function can make a nice visual representation. Here is an example using some fictions data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>)
<span class="kw">names</span>(x) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bob&quot;</span>, <span class="st">&quot;Alice&quot;</span>, <span class="st">&quot;Jane&quot;</span>, <span class="st">&quot;Fred&quot;</span>, <span class="st">&quot;Barney&quot;</span>, <span class="st">&quot;Lucy&quot;</span>)
x</code></pre></div>
<pre><code>##    Bob  Alice   Jane   Fred Barney   Lucy 
##      2      3      6      3      2      4</code></pre>
<p>From this, a barplot can be constructed where the x-axis has discrete entities for each name and the y-axis represents the magnitude of whatever it is we are measuring.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">barplot</span>(x, <span class="dt">xlab =</span> <span class="st">&quot;People&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Thinn-A-Ma-Jigs&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-141-1.png" width="672" /></p>
<p>Notice here that I included labels for both the x- and y-axes. You do this by inserting these arguments into the parameters passed to the <code>barplot()</code> (it is the same for all built-in plotting as we will see below).</p>
<p>To use <code>ggplot</code> for displaying this, you have to present the data in a slightly different way. There are shortcuts using the <code>qplot()</code> function but I prefer to use the more verbose approach. Here is how we would plot the same output using ggplot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Thinn_A_Ma_Jigs=</span>x,<span class="dt">names=</span><span class="kw">names</span>(x))
<span class="kw">ggplot</span>( df, <span class="kw">aes</span>(<span class="dt">x=</span>names,<span class="dt">y=</span>Thinn_A_Ma_Jigs)) +<span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-142-1.png" width="672" /></p>
<p>There are a couple of things to point out about this.</p>
<ol style="list-style-type: decimal">
<li>This is a compound statement, the plotting commands are literally ‘added together’.</li>
<li>The <code>ggplot()</code> function specifies a <code>data.frame</code> from which your data will be extracted.</li>
<li>The <code>aes()</code> component within the <code>ggplot()</code> function describes the aesthetics of the plot. This is how you designate which variable columns in the <code>data.frame</code> will be on the x- and y- axes, colors, plot shapes, categories, etc.</li>
<li>To the basic description of the data and aesthetics, a geometry is added, that of a barplot. If necessary, you can pass additional information to the <code>geom_bar()</code> function. In this case, I had to tell it that you use the raw data <code>stat='identity'</code> instead of trying to summarize many observations.<br />
</li>
<li>The axis labels are taken directly from the names of the columns in the <code>data.frame</code>.</li>
</ol>
<p>If the data on the x-axis is not discrete but measured on a single variable, then the <code>hist()</code> function can take the data and categorize it into groups based upon the density of the underlying data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">h &lt;-<span class="st"> </span><span class="kw">hist</span>(iris$Sepal.Length)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-143-1.png" width="672" /></p>
<p>The <code>hist()</code> function itself returns a bit of information that may be of interest. It is not necessary that you capture these data, R will ignore it if you do not assign it to a variable. What is of interest though is that it returns a <code>list()</code> object with the following keys:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(h)</code></pre></div>
<pre><code>## [1] &quot;breaks&quot;   &quot;counts&quot;   &quot;density&quot;  &quot;mids&quot;     &quot;xname&quot;    &quot;equidist&quot;</code></pre>
<p>The data within these keys can be accessed directly as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">h</code></pre></div>
<pre><code>## $breaks
## [1] 4.0 4.5 5.0 5.5 6.0 6.5 7.0 7.5 8.0
## 
## $counts
## [1]  5 27 27 30 31 18  6  6
## 
## $density
## [1] 0.06666667 0.36000000 0.36000000 0.40000000 0.41333333 0.24000000
## [7] 0.08000000 0.08000000
## 
## $mids
## [1] 4.25 4.75 5.25 5.75 6.25 6.75 7.25 7.75
## 
## $xname
## [1] &quot;iris$Sepal.Length&quot;
## 
## $equidist
## [1] TRUE
## 
## attr(,&quot;class&quot;)
## [1] &quot;histogram&quot;</code></pre>
<p>and provide the basic information on how to construct the histogram as shown below. The values defined for the break locations for each of the bars and the midpoints are the default ones for the <code>hist()</code> function. If you wish, you can provide the number of breaks in the data, the location of the breaks, including the lowest and highest terms, plot as a count versus a frequency, etc.</p>
<p>The corresponding <code>ggplot</code> approach is</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length)) +<span class="st"> </span><span class="kw">geom_bar</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-146-1.png" width="672" /></p>
<p>which also takes a few liberties with the default binning parameters.</p>
<p>By default, both the R and <code>ggplot</code> approches make a stab at creating enough of the meta data around the plot to make it somewhat useful. However, it is helpful if we add a bit to that and create graphics that suck just a little bit less. There are many options that we can add to the standard plot command to make it more informative and/or appropriate for your use. The most common ones include:</p>
<ul>
<li><code>xlab</code> Change the label on the x-axis<br />
</li>
<li><code>ylab</code> Change the lable on the y-axis</li>
<li><code>main</code> Change the title of the graph</li>
<li><code>col</code> Either a single value, or a vector of values, indicating the color to be plot.</li>
<li><code>xlim/ylim</code> The limits for the x- and y-axes.</li>
</ul>
<p>Additional options are given in tabular form in the next section.</p>
<p>Here is another plot of the same data but spiffied up a bit.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">hist</span>( iris$Sepal.Length,<span class="dt">xlab=</span><span class="st">&quot;Sepal Length (cm)&quot;</span>, 
           <span class="dt">ylab=</span><span class="st">&quot;Count&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>, <span class="dt">col=</span><span class="st">&quot;lightblue&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-147-1.png" width="672" /></p>
<p>A few things to notice:</p>
<ol style="list-style-type: decimal">
<li>The main title over the top of the image was visually removed by assigning an empty characters string to it.<br />
</li>
<li>The color parameter refers to the fill color of the bars, not the border color.</li>
</ol>
<p>In <code>ggplot</code> we add those additional components either to the geometry where it is defined (e.g., the color in the geom_bar) or to the overall plot (as in the addition of labels on the axes).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length)) +<span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">fill=</span><span class="st">&quot;lightblue&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Sepal Length (cm)&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-148-1.png" width="672" /></p>
</div>
<div id="plotting-parameters" class="section level2">
<h2><span class="header-section-number">5.2</span> Plotting Parameters</h2>
<p>There are several parameters common to the various plotting functions used in the basic R plotting functions. Here is a short table of the most commonly used ones.</p>
<table style="width:100%;">
<colgroup>
<col width="12%" />
<col width="24%" />
<col width="63%" />
</colgroup>
<thead>
<tr class="header">
<th>Command</th>
<th>Usage</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>bg</td>
<td>bg=“white”</td>
<td>Sets the background color for the entire figure.</td>
</tr>
<tr class="even">
<td>bty</td>
<td>bty=“n”</td>
<td>Sets the style of the box type around the graph. Useful values are “o” for complete box (the default), “l”, “7”, “c”, “u”, “]” which will make a box with sides around the plot area resembling the upper case version of these letters, and “n” for no box.</td>
</tr>
<tr class="odd">
<td>cex</td>
<td>cex=2</td>
<td>Magnifies the default font size by the corresponding factor.</td>
</tr>
<tr class="even">
<td>cex.axis</td>
<td>cex.axis=2</td>
<td>Magnifies the font size on the axes.</td>
</tr>
<tr class="odd">
<td>col</td>
<td>col=“blue”</td>
<td>Sets the plot color for the points, lines, etc.</td>
</tr>
<tr class="even">
<td>fg</td>
<td>fg=“blue”</td>
<td>Sets the foreground color for the image.</td>
</tr>
<tr class="odd">
<td>lty</td>
<td>lty=1</td>
<td>Sets the type of line as 0-none, 1-solid, 2-dashed, 3-dotted, etc.</td>
</tr>
<tr class="even">
<td>lwd</td>
<td>lwd=3</td>
<td>Specifies the width of the line.</td>
</tr>
<tr class="odd">
<td>main</td>
<td>main=“title”</td>
<td>Sets the title to be displayed over the top of the graph.</td>
</tr>
<tr class="even">
<td>mfrow</td>
<td>mfrow=c(2,2)</td>
<td>Creates a ‘matrix’ of plots in a single figure.</td>
</tr>
<tr class="odd">
<td>pch</td>
<td>pch=16</td>
<td>Sets the type of symbol to be used in a scatter plot.</td>
</tr>
<tr class="even">
<td>sub</td>
<td>sub=“subtitle”</td>
<td>Sets the subtitle under the main title on the graph.</td>
</tr>
<tr class="odd">
<td>type</td>
<td>type=“l”</td>
<td>Specifies the type of the graph to be shown using a generic plot() command. Types are “p” for points (default), “l” for lines, and “b” for both.</td>
</tr>
<tr class="even">
<td>xlab</td>
<td>xlab=“Size (m)”</td>
<td>Sets the label attached to the x-axis.</td>
</tr>
<tr class="odd">
<td>ylab</td>
<td>ylab=“Frequency”</td>
<td>Sets the label attached to the y-axis.</td>
</tr>
</tbody>
</table>
</div>
<div id="density-plots" class="section level2">
<h2><span class="header-section-number">5.3</span> Density Plots</h2>
<p>Another way of looking at the distribution of univariate data is through the use of the <code>density()</code> function. This function takes a vector of values and creates a probability density function of it returning an object of class “density”.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d &lt;-<span class="st"> </span><span class="kw">density</span>( iris$Sepal.Length) 
<span class="kw">attributes</span>(d)</code></pre></div>
<pre><code>## $names
## [1] &quot;x&quot;         &quot;y&quot;         &quot;bw&quot;        &quot;n&quot;         &quot;call&quot;      &quot;data.name&quot;
## [7] &quot;has.na&quot;   
## 
## $class
## [1] &quot;density&quot;</code></pre>
<p>The <code>density()</code> function takes the univariate data and fits its density (internally represented by <code>d$y</code>) along the range of values (in <code>d$x</code>).</p>
<p>A plot can be produced using d as the variables as follows (with a bit of extra plotting parameters as depicted in the previous table).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>( d, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">lwd =</span> <span class="dv">2</span>, 
      <span class="dt">xlab =</span> <span class="st">&quot;Value of X&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Frequency&quot;</span>, 
      <span class="dt">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;Density Plot&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-150-1.png" width="672" /></p>
<p>The corresponding <code>ggplot</code> approach is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length)) +<span class="st"> </span><span class="kw">geom_density</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-151-1.png" width="672" /></p>
</div>
<div id="overlaying-plots" class="section level2">
<h2><span class="header-section-number">5.4</span> Overlaying Plots</h2>
<p>The data in the previous density plot represents the sepal lengths across all three iris species. It may be more useful to plot them as the density of each species instead of combined. Overlaying plots is a pretty easy feature built into the default R plotting functionalities.</p>
<p>Lets look at the data and see if the mean length differs across species. I do this using the <code>by()</code> command, which takes three parameters; the first parameter is the raw data you want to examine, the second one is the way you would like to partition that data, and the third one is the function you want to call on those groupings of raw data. This general form</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">by</span>( data, grouping, function )</code></pre></div>
<p>mixing both data and functions to be used, is pretty common in R and you will see it over and over again. The generic call below asks to take the sepal data and partition it by species and estimate the mean.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">by</span>( iris$Sepal.Length, iris$Species, mean)</code></pre></div>
<pre><code>## iris$Species: setosa
## [1] 5.006
## -------------------------------------------------------- 
## iris$Species: versicolor
## [1] 5.936
## -------------------------------------------------------- 
## iris$Species: virginica
## [1] 6.588</code></pre>
<p>So there may be differences. Lets pull the data out and create density plots for each.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d.setosa &lt;-<span class="st"> </span>iris$Sepal.Length[ iris$Species==<span class="st">&quot;setosa&quot;</span> ]
d.versicolor &lt;-<span class="st"> </span>iris$Sepal.Length[ iris$Species==<span class="st">&quot;versicolor&quot;</span> ]
d.virginica &lt;-<span class="st"> </span>iris$Sepal.Length[ iris$Species==<span class="st">&quot;virginica&quot;</span> ]
d.se &lt;-<span class="st"> </span><span class="kw">density</span>( d.setosa )
d.ve &lt;-<span class="st"> </span><span class="kw">density</span>( d.versicolor )
d.vi &lt;-<span class="st"> </span><span class="kw">density</span>( d.virginica )</code></pre></div>
<p>I can now plot the densities independently. After the first <code>plot()</code> function is called, you can add to it by using <code>lines()</code> or <code>points()</code> function calls. They will overlay the subsequent plots over the first one. One of the things you need to be careful of is that you need to make sure the x- and y-axes are properly scaled so that subsequent calls to <code>points()</code> or <code>lines()</code> does not plot stuff outside the boundaries of your initial plot.</p>
<p>Here I plot the setosa data first, specify the <code>xlim</code> (limits of the x-axis), set the color, and labels. On subsequent plotting calls, I do not specify labels but set alternative colors. Then a nice legend is placed on the graph, the coordinates of which are specified on the values of the x- and y-axis (I also dislike the boxes around graphic so I remove them as well with <code>bty=&quot;n&quot;</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(d.se,<span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">8</span>),<span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, 
     <span class="dt">xlab=</span><span class="st">&quot;Sepal Length (cm)&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Sepal Lengths&quot;</span>)
<span class="kw">lines</span>( d.ve, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">8</span>), <span class="dt">col=</span><span class="st">&quot;green&quot;</span>,<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)
<span class="kw">lines</span>( d.vi, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">8</span>), <span class="dt">col=</span><span class="st">&quot;blue&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)
<span class="kw">legend</span>( <span class="fl">6.5</span>,<span class="fl">1.1</span>,<span class="kw">c</span>(<span class="st">&quot;I. setosa&quot;</span>, <span class="st">&quot;I. versicolor&quot;</span>, <span class="st">&quot;I. virginica&quot;</span>), 
        <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;blue&quot;</span>), <span class="dt">lwd=</span><span class="dv">2</span>,<span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-155-1.png" width="672" /></p>
<p>A lot of that background material is unnecessary using <code>geom_density()</code> because we can specify to the plotting commands that the data are partitioned by the values contained in the <code>Species</code> column of the <code>data.frame</code>. This allows us to make this plot</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(Sepal.Length,<span class="dt">color=</span>Species)) +<span class="st"> </span><span class="kw">geom_density</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-156-1.png" width="672" /></p>
<p>or a correpsonding one using <code>fill=Species</code> instead (I set the alpha transparency to allow you to see the plots in the background).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(Sepal.Length,<span class="dt">fill=</span>Species)) +<span class="st"> </span><span class="kw">geom_density</span>( <span class="dt">alpha=</span><span class="fl">0.8</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-157-1.png" width="672" /></p>
<p>We can use the <code>barplot()</code> function here as well and either stack or stagger the density of sepal lengths using discrete bars.</p>
<p>Here I make a matrix of bins using the <code>hist()</code> function with a specified set of breaks and then use it to plot discrete bin counts using the <code>barplot()</code> function. I include this example here because there are times when you want to produce a stacked bar plot (rarely) or a staggered barplot (more common) for some univariate data source and I always forget how to specifically do that.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">breaks &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="dv">4</span>,<span class="dv">8</span>,<span class="dt">by=</span><span class="fl">0.2</span>)
h.se &lt;-<span class="st"> </span><span class="kw">hist</span>(d.setosa, <span class="dt">breaks=</span>breaks, <span class="dt">plot =</span> <span class="ot">FALSE</span>)
h.ve &lt;-<span class="st"> </span><span class="kw">hist</span>(d.versicolor, <span class="dt">breaks=</span>breaks, <span class="dt">plot=</span><span class="ot">FALSE</span>)
h.vi &lt;-<span class="st"> </span><span class="kw">hist</span>(d.virginica, <span class="dt">breaks=</span>breaks, <span class="dt">plot=</span><span class="ot">FALSE</span>)
vals &lt;-<span class="st"> </span><span class="kw">rbind</span>( h.se$counts, h.ve$counts, h.vi$counts )
<span class="kw">rownames</span>(vals) &lt;-<span class="st"> </span><span class="kw">levels</span>(iris$Species)
<span class="kw">colnames</span>(vals) &lt;-<span class="st"> </span>breaks[<span class="dv">1</span>:<span class="dv">20</span>]
vals</code></pre></div>
<pre><code>##            4 4.2 4.4 4.6 4.8  5 5.2 5.4 5.6 5.8 6 6.2 6.4 6.6 6.8 7 7.2
## setosa     0   4   5   7  12 11   6   2   3   0 0   0   0   0   0 0   0
## versicolor 0   0   0   0   3  2   1  10   8   6 6   5   3   4   2 0   0
## virginica  0   0   0   0   1  0   0   1   4   3 4  11   4   7   3 4   2
##            7.4 7.6 7.8
## setosa       0   0   0
## versicolor   0   0   0
## virginica    1   4   1</code></pre>
<p>The matrix of data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">barplot</span>(vals,<span class="dt">xlab=</span><span class="st">&quot;Sepal Length&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Frequency&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-159-1.png" width="672" /></p>
<p>Stacked barplots may or may not be that informative, depending upon the complexity of your underlying data. It is helpful though to be able to stagger them. In the basic <code>barplot()</code> function allows you to specify the bars to be spaced beside each other as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">barplot</span>(vals,<span class="dt">xlab=</span><span class="st">&quot;Sepal Length&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Frequency&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;blue&quot;</span>), <span class="dt">beside=</span><span class="ot">TRUE</span>)
<span class="kw">legend</span>(<span class="dv">60</span>, <span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;I. setosa&quot;</span>, <span class="st">&quot;I. versicolor&quot;</span>, <span class="st">&quot;I. virginica&quot;</span>), <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-160-1.png" width="672" /></p>
<p>For plotting onto a barplot object, the x-axis number is not based upon the labels on the x-axis. It is an integer that relates to the number of bar-widths and separations. In this example, there are three bars for each category plus one separator (e.g., the area between two categories). So I had to plot the legend at 60 units on the x-coordinate, which puts it just after the 15th category (e.g., 15*4=60).</p>
<p>We can do the same thing with <code>geom_histogram()</code>, though again with a bit less typing involved. Here is the raw plot (which by default stacks just like in the <code>barplot()</code> example)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length, <span class="dt">fill=</span>Species)) +<span class="st"> </span><span class="kw">geom_histogram</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-161-1.png" width="672" /></p>
<p>and the correspondingly staggered plot with bars positioned next to eachother.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length, <span class="dt">fill=</span>Species)) +<span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">position=</span><span class="st">&quot;dodge&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-162-1.png" width="672" /></p>
<p>Notice how we add the fill color to the <code>aes()</code> function because the categories for assigning colors will be extracted from the <code>data.frame</code>, whereas when we just wanted to set them all light blue, the color is specified outside the <code>aes()</code> function. Also notice when we specify something to the <code>aes()</code> command in this way, we do not quote the name of the column, we call it just as if it were a normal variable.</p>
</div>
<div id="boxplots" class="section level2">
<h2><span class="header-section-number">5.5</span> Boxplots</h2>
<p>Boxplots are a display of distributions of continuous data within categories, much like what we displayed in the previous staggered barplot. It is often the case that boxplots depict differences in mean values while giving some indication of the dispersal around that mean. This can be done in the staggered barplot as above but not quite as specifically.</p>
<p>The default display for boxplots in R provides the requires a factor and a numeric data type. The factor will be used as the categories for the x-axis on which the following visual components will be displayed:</p>
<ul>
<li>The median of the data for each factor,<br />
</li>
<li>An optional 5% confidence interval (shown by passing notch=TRUE) around the mean.</li>
<li>The inner 50th percentile of the data,<br />
</li>
<li>Outlier data points.</li>
</ul>
<p>Here is how the sepal length can be plot as a function of the Iris species.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(Sepal.Length ~<span class="st"> </span>Species, <span class="dt">data=</span>iris, <span class="dt">notch=</span><span class="ot">TRUE</span>, 
        <span class="dt">xlab=</span><span class="st">&quot;Species&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Sepal Length (cm)&quot;</span>, 
        <span class="dt">frame.plot=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-163-1.png" width="672" /></p>
<p>A couple of things should be noted on this plot call. First, I used a ‘functional’ form for the relationship between the continuous variable (<code>Sepal.Length</code>) and the factor (<code>Species</code>) with the response variable indicated on the left side of the tilde and the predictor variables on the right side. It is just as reasonable to use <code>boxplot(Species,Sepal.Length, ...)</code> in the call as x- and y- variables in the first two positions. However, it reads a bit better to do it as a function, like a regression. Also, it should be noted that I added the optional term, <code>data=iris</code>, to the call. This allowed me to reference the columns in the iris data.frame without the dollar sign notation. Without that I would have had to write <code>boxplot( data$Sepal.Length ~ data$Species, ...)</code>, a much more verbose way of doing it but most programmers are relatively lazy and anything they can do to get more functionality out of less typing… Finally, I set <code>frame.plot=FALSE</code> to remove the box around the plot since the <code>bty=&quot;n&quot;</code> does not work on boxplots (big shout out for consistency!). I don’t know why these are different, I just hate the box.</p>
<p>The corresponding <code>ggplot</code> approach is the same as the <code>boxplot()</code> one, we just have to specify the <code>notch=TRUE</code> in the <code>geom_boxplot()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(<span class="dt">x=</span>Species, <span class="dt">y=</span>Sepal.Length)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">notch =</span> <span class="ot">TRUE</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Iris Species&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Sepal Width (cm)&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-164-1.png" width="672" /></p>
</div>
<div id="scatter-plots" class="section level2">
<h2><span class="header-section-number">5.6</span> Scatter Plots</h2>
<p>With two sets of continuous data, we can produce scatter plots. These consist of 2- (or 3-) coordinates where we can plot our data. With the addition of alternative colors (<code>col=</code>) and/or plot shapes (<code>pch=</code>) passed to the generic plot() command, we can make really informative graphical output. Here I plot two characteristics of the iris data set, sepal length and width, and use the species to indicate alternative symbols.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>( iris$Sepal.Length, iris$Sepal.Width, 
      <span class="dt">pch=</span><span class="kw">as.numeric</span>(iris$Species), <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, 
      <span class="dt">xlab=</span><span class="st">&quot;Sepal Length (cm)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Sepal Width (cm)&quot;</span>)
<span class="kw">legend</span>( <span class="fl">6.75</span>, <span class="fl">4.3</span>, <span class="kw">c</span>(<span class="st">&quot;I. setosa&quot;</span>, <span class="st">&quot;I. versicolor&quot;</span>, <span class="st">&quot;I. virginica&quot;</span>), 
        <span class="dt">pch=</span><span class="dv">1</span>:<span class="dv">3</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-165-1.png" width="672" /></p>
<p>The call to <code>pch=</code> that I used coerced a factor into a numeric data type. By default, this will create a numeric sequence (starting at 1) for all levels of the factor, including ones that may not be in the data you are doing the conversion on. The parameter passed to <code>pch</code> is an integer that determines the shape of the symbol being plot. I typically forget which numbers correspond to which symbols (there are 25 of them in total ) and have to look them up when I need them. One of the easiest ways is just to plot them as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dv">1</span>:<span class="dv">25</span>,<span class="dv">1</span>:<span class="dv">25</span>,<span class="dt">pch=</span><span class="dv">1</span>:<span class="dv">25</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;pch&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;pch&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-166-1.png" width="672" /></p>
<p>which produces the following figure where you can choose the appropriate symbols for your plotting needs.</p>
<p>A scatter plot in <code>ggplot</code> is created using the <code>geom_point()</code> layer.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(iris,<span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length,<span class="dt">y=</span>Sepal.Width, <span class="dt">shape=</span>Species)) +<span class="st"> </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-167-1.png" width="672" /></p>
<p>and the shapes are specified automatically.</p>
</div>
<div id="multiple-plots" class="section level2">
<h2><span class="header-section-number">5.7</span> Multiple Plots</h2>
<p>The last figure was a bit confusing. You can see a few points where the three species have the same measurements for both sepal length and width—both <em>I. versicolor</em> and <em>I. virginica</em> have examples with sepal widths of 3.0cm and lengths of 6.7cm. For some of these it would be difficult to determine using a graphical output like this if there were more overlap. In this case, it may be a more informative approach to make plots for each species rather than using different symbols.</p>
<p>In <code>ggplot</code>, there is a facet geometry that can be added to a plot that can pull apart the individual species plots (in this case) and plot them either next to each other (if you are looking at y-axis differences), on top of each other (for x-axis comparisons), or as a grid (for both x- and y- axis comparisons). This layer is added to the plot using <code>facet_grid()</code> and take a functional argument (as I used for the boxplot example above) on which factors will define rows and columns of plots. Here is an example where I stack plots by species.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( iris, <span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Width, <span class="dt">y=</span>Sepal.Length ) ) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>() +<span class="st"> </span><span class="kw">facet_grid</span>(Species~.) +<span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Sepal Width (cm)&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Sepal Length&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-168-1.png" width="672" /></p>
<p>The <code>Species ~ .</code> means that the rows will be defined by the levels indicatd in <code>iris$Species</code> and all the plotting (the period part) will be done as columns. If you have more than one factor, you can specify <code>RowFactor ~ ColumnFactor</code> and it will make the corresponding grid.</p>
<p>I would argue that this is a much more intuitive display of differences in width than the previous plot.</p>
<p>You can achieve a similar effect using built-in plotting functions as well. To do this, we need to mess around a bit with the plotting attributes. These are default parameters (hence the name <code>par</code>) for plotting that are used each time we make a new graphic. Here are all of them (consult with the previous Table for some of the meanings).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>( <span class="kw">par</span>() )</code></pre></div>
<pre><code>##  [1] &quot;xlog&quot;      &quot;ylog&quot;      &quot;adj&quot;       &quot;ann&quot;       &quot;ask&quot;      
##  [6] &quot;bg&quot;        &quot;bty&quot;       &quot;cex&quot;       &quot;cex.axis&quot;  &quot;cex.lab&quot;  
## [11] &quot;cex.main&quot;  &quot;cex.sub&quot;   &quot;cin&quot;       &quot;col&quot;       &quot;col.axis&quot; 
## [16] &quot;col.lab&quot;   &quot;col.main&quot;  &quot;col.sub&quot;   &quot;cra&quot;       &quot;crt&quot;      
## [21] &quot;csi&quot;       &quot;cxy&quot;       &quot;din&quot;       &quot;err&quot;       &quot;family&quot;   
## [26] &quot;fg&quot;        &quot;fig&quot;       &quot;fin&quot;       &quot;font&quot;      &quot;font.axis&quot;
## [31] &quot;font.lab&quot;  &quot;font.main&quot; &quot;font.sub&quot;  &quot;lab&quot;       &quot;las&quot;      
## [36] &quot;lend&quot;      &quot;lheight&quot;   &quot;ljoin&quot;     &quot;lmitre&quot;    &quot;lty&quot;      
## [41] &quot;lwd&quot;       &quot;mai&quot;       &quot;mar&quot;       &quot;mex&quot;       &quot;mfcol&quot;    
## [46] &quot;mfg&quot;       &quot;mfrow&quot;     &quot;mgp&quot;       &quot;mkh&quot;       &quot;new&quot;      
## [51] &quot;oma&quot;       &quot;omd&quot;       &quot;omi&quot;       &quot;page&quot;      &quot;pch&quot;      
## [56] &quot;pin&quot;       &quot;plt&quot;       &quot;ps&quot;        &quot;pty&quot;       &quot;smo&quot;      
## [61] &quot;srt&quot;       &quot;tck&quot;       &quot;tcl&quot;       &quot;usr&quot;       &quot;xaxp&quot;     
## [66] &quot;xaxs&quot;      &quot;xaxt&quot;      &quot;xpd&quot;       &quot;yaxp&quot;      &quot;yaxs&quot;     
## [71] &quot;yaxt&quot;      &quot;ylbias&quot;</code></pre>
<p>To create multiple plots on one graphic, we need to modify either the mfrow or mfcol property (either will do). They represent the number of figures to plot as a 2-integer vector for rows and columns. By default, it is set to</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>()$mfrow</code></pre></div>
<pre><code>## [1] 1 1</code></pre>
<p>because there is only 1 row and 1 column in the plot. To change this, we simply pass a new value to the <code>par()</code> command and then do the plotting. In the following example, I plot the results of a linear model (<code>lm()</code>) function call. This returns four plots looking at the normality of the data, residuals, etc. This time, instead of seeing them one at a time, I’m going to plot all four into one figure.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))
<span class="kw">plot</span>( <span class="kw">lm</span>( Sepal.Length ~<span class="st"> </span>Sepal.Width, <span class="dt">data=</span>iris))</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-171-1.png" width="672" /></p>
</div>
<div id="color-palettes" class="section level2">
<h2><span class="header-section-number">5.8</span> Color Palettes</h2>
<p>The default plotting colors in R are black and white. However, there is a rich set of colors available for your plotting needs. The easiest set are named colors. At the time of this writing, there are</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">colors</span>())</code></pre></div>
<pre><code>## [1] 657</code></pre>
<p>different colors available for your use. Not all are distinct as some overlap. However the benefit of these colors is that they have specific names, making it easier for you to remember than RGB or hex representations. Here are a random set of 20 color names.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colors</span>()[ <span class="kw">sample.int</span>( <span class="kw">length</span>(<span class="kw">colors</span>()), <span class="dt">size=</span><span class="dv">20</span>) ]</code></pre></div>
<pre><code>##  [1] &quot;orangered4&quot;     &quot;lightpink4&quot;     &quot;red2&quot;           &quot;gray32&quot;        
##  [5] &quot;gray87&quot;         &quot;paleturquoise4&quot; &quot;gray78&quot;         &quot;gray80&quot;        
##  [9] &quot;palegreen1&quot;     &quot;slategray1&quot;     &quot;lightpink1&quot;     &quot;gray86&quot;        
## [13] &quot;orchid1&quot;        &quot;grey27&quot;         &quot;azure1&quot;         &quot;beige&quot;         
## [17] &quot;grey70&quot;         &quot;grey39&quot;         &quot;lightskyblue4&quot;  &quot;darkgoldenrod&quot;</code></pre>
<p>To use these colors you can call them by name in the col= option to a plot. Here is an example where I define three named colors and then coerce the <code>iris$Species</code> variable into an integer to select the color by species and plot it in a scatter plot (another version of the <code>pch=</code> example previously).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;royalblue1&quot;</span>, <span class="st">&quot;orange1&quot;</span>, <span class="st">&quot;green3&quot;</span>)
cols &lt;-<span class="st"> </span>colors[ iris$Species ]
<span class="kw">plot</span>( Sepal.Width ~<span class="st"> </span>Sepal.Length, <span class="dt">data=</span>iris, 
      <span class="dt">col=</span>cols, <span class="dt">xlab=</span><span class="st">&quot;Sepal Length (cm)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Sepal Width (cm)&quot;</span>, 
      <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">pch=</span><span class="dv">16</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-174-1.png" width="672" /></p>
<p>There is a lot of colors to choose from, and you are undoubtedly able to fit any color scheme on any presentation you may encounter.</p>
<p>In addition to named colors, there are color palettes available for you to grab the hex value for individual color along a pre-defined gradient. These colors ramps are:</p>
<ul>
<li><code>rainbow()</code>: A palette covering the visible spectrum<br />
</li>
<li><code>heat.colors()</code>: A palette ranging from red, through orange and yellow, to white.<br />
</li>
<li><code>terrain.colors()</code>: A palette for plotting topography with lower values as green and increasing through yellow, brown, and finally white.<br />
</li>
<li><code>topo.colors()</code>: A palette starting at dark blue (water) and going through green (land) and yellow to beige (mountains).<br />
</li>
<li><code>cm.colors()</code>: A palette going from light blue through white to pink.</li>
</ul>
<p>and are displayed in the following figure.</p>
<p>The individual palette functions return the hex value for an equally separated number of colors along the palette, you only need to ask for the number of colors you are requesting. Here is an example from the <code>rainbow()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rainbow</span>(<span class="dv">10</span>)</code></pre></div>
<pre><code>##  [1] &quot;#FF0000FF&quot; &quot;#FF9900FF&quot; &quot;#CCFF00FF&quot; &quot;#33FF00FF&quot; &quot;#00FF66FF&quot;
##  [6] &quot;#00FFFFFF&quot; &quot;#0066FFFF&quot; &quot;#3300FFFF&quot; &quot;#CC00FFFF&quot; &quot;#FF0099FF&quot;</code></pre>
</div>
<div id="rcolorbrewer" class="section level2">
<h2><span class="header-section-number">5.9</span> RColorBrewer</h2>
<p>The ggplot2 library has a slightly more interesting color palette, but it too is often a bit lacking. You have total control over the colors you produce and can specify them as RGB, hex (e.g. the way the web makes colors), or CMYK. You can also define your own color palettes.</p>
<p>To start, the RColorBrewer library provides several palettes for plotting either quantitative, qualitative, or divergent data. These three groups and the names of individual color palettes can be viewed as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(RColorBrewer)
<span class="kw">display.brewer.all</span>()</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-176-1.png" width="672" /></p>
<p>In ggplot, we can use these palettes as follows.</p>
<p>You can change the default palette in ggplot output adding the <code>scale_color_brewer()</code> (and <code>scale_fill_brewer()</code> if you are doing fill colors like in a barplot) to the plot. Here is an example where I change the default palette to the 6th divergent palette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(iris,<span class="kw">aes</span>(<span class="dt">x=</span>Sepal.Length,<span class="dt">y=</span>Sepal.Width, <span class="dt">color=</span>Species)) +<span class="st"> </span><span class="kw">geom_point</span>() 
p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Sepal Length (cm)&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Sepal Width (cm)&quot;</span>)
p +<span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">type=</span><span class="st">&quot;div&quot;</span>, <span class="dt">palette=</span><span class="dv">6</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-177-1.png" width="672" /></p>
</div>
<div id="saving-imagery" class="section level2">
<h2><span class="header-section-number">5.10</span> Saving Imagery</h2>
<p>Creating a graphic for display on your screen is only the first step. For it to be truly useful, you need to save it to a file, often with specific parameters such as DPI or image size, so that you can use it outside of R.</p>
<p>For normal R graphics, you can copy the current graphical device to a file using <code>dev.copy()</code> or you can create a graphic file object and plot directly to it instead of the display. In both cases, you will need to make sure that you have your graphic formatted the way you like. Once you have the file configured the way you like, decide on the image format you will want. Common ones are jpg, png and tiff. Here are the function arguments for each of the formats.</p>
<pre><code>jpeg(filename = &quot;Rplot%03d.jpeg&quot;,
     width = 480, height = 480, units = &quot;px&quot;, pointsize = 12,
     quality = 75,
     bg = &quot;white&quot;, res = NA, ...,
     type = c(&quot;cairo&quot;, &quot;Xlib&quot;, &quot;quartz&quot;), antialias)

png(filename = &quot;Rplot%03d.png&quot;,
    width = 480, height = 480, units = &quot;px&quot;, pointsize = 12,
     bg = &quot;white&quot;,  res = NA, ...,
    type = c(&quot;cairo&quot;, &quot;cairo-png&quot;, &quot;Xlib&quot;, &quot;quartz&quot;), antialias)

tiff(filename = &quot;Rplot%03d.tiff&quot;,
     width = 480, height = 480, units = &quot;px&quot;, pointsize = 12,
     compression = c(&quot;none&quot;, &quot;rle&quot;, &quot;lzw&quot;, &quot;jpeg&quot;, &quot;zip&quot;, &quot;lzw+p&quot;, &quot;zip+p&quot;),
     bg = &quot;white&quot;, res = NA,  ...,
     type = c(&quot;cairo&quot;, &quot;Xlib&quot;, &quot;quartz&quot;), antialias)</code></pre>
<p>These are the default values. If you need a transparent background, it is easiest to use the png file and set <code>bg=&quot;transparent&quot;</code>. If you are producing an image for a publication, you will most likely be given specific parameters about the image format in terms of DPI, pointsize, dimensions, etc. They can be set here.</p>
<p>To create the image, call the appropriate function above, and a file will be created for your plotting. You must then call the plotting functions to create your graphic. Instead of showing up in a graphical display window, they will instead be plot to the file. When done you must tell R that your plotting is now finished and it can close the graphics file. This is done by calling dev.off(). Here is an example workflow saving an image of a scatter plot with a smoothed line through it using ggplot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">png</span>( <span class="dt">filename =</span> <span class="st">&quot;MyCoolGraphic.png&quot;</span>, <span class="dt">quality=</span><span class="dv">300</span>, <span class="dt">bg=</span><span class="st">&quot;transparent&quot;</span>, <span class="dt">width=</span><span class="dv">1080</span>, <span class="dt">height=</span><span class="dv">1080</span>)
<span class="kw">ggplot</span>( df, <span class="kw">aes</span>(<span class="dt">x=</span>PredictorVar, <span class="dt">y=</span>ResponseVar)) +<span class="st"> </span><span class="kw">geom_point</span>() +<span class="st"> </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;loess&quot;</span>) +<span class="st"> </span><span class="kw">theme_bw</span>()
<span class="kw">dev.off</span>()</code></pre></div>
<p>This will create a file MyCoolGraphic.png saved in the same folder as the current working directory.</p>
<p>To use <code>dev.copy()</code> instead, you first make the graphic and then copy the current graphic to a file. For this example, it would be:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>( df, <span class="kw">aes</span>(<span class="dt">x=</span>PredictorVar, <span class="dt">y=</span>ResponseVar)) +<span class="st"> </span><span class="kw">geom_point</span>() +<span class="st"> </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;loess&quot;</span>) +<span class="st"> </span><span class="kw">theme_bw</span>()
<span class="kw">dev.copy</span>(<span class="dt">device=</span>png,<span class="dt">file=</span><span class="st">&quot;MyDevCopyGraphic.png&quot;</span>)
<span class="kw">dev.off</span>()</code></pre></div>
<p>In general, I prefer the first method as it allows me to specify the specifics of the file in an easier fashion than the second approach.</p>
<p>If you are using ggplot2 graphics, there is a built-in function ggsave() that can also be used to save your currently displaying graphic (or others if you so specify) to file. Here are the specifics on that function call.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="dt">filename =</span> <span class="kw">default_name</span>(plot), <span class="dt">plot =</span> <span class="kw">last_plot</span>(),
  <span class="dt">device =</span> <span class="kw">default_device</span>(filename), <span class="dt">path =</span> <span class="ot">NULL</span>, <span class="dt">scale =</span> <span class="dv">1</span>,
  <span class="dt">width =</span> <span class="kw">par</span>(<span class="st">&quot;din&quot;</span>)[<span class="dv">1</span>], <span class="dt">height =</span> <span class="kw">par</span>(<span class="st">&quot;din&quot;</span>)[<span class="dv">2</span>], <span class="dt">units =</span> <span class="kw">c</span>(<span class="st">&quot;in&quot;</span>, <span class="st">&quot;cm&quot;</span>,
  <span class="st">&quot;mm&quot;</span>), <span class="dt">dpi =</span> <span class="dv">300</span>, <span class="dt">limitsize =</span> <span class="ot">TRUE</span>, ...)</code></pre></div>
<p>Either approach will allow you to produce publication quality graphics.</p>
</div>
<div id="interactive-graphics" class="section level2">
<h2><span class="header-section-number">5.11</span> Interactive Graphics</h2>
<p>There are several other libraries available for creating useful plots in R. The most dynamical ones are based on javascript and as such are restricted to web or ebook displays. Throughout this book, I will be showing how you can use various libraries to create interactive content as well as providing some ‘widgets’ that will demonstrate how some feature of popualtion or landscape structure changes under a specified model. These widgets are also written in R and are hosted online as a <a href="http://shiny.rstudio.org">shiny</a> application. I would encourage you to look into creating dynamcial reports using these techniques, they <em>can</em> facilitate a better understanding of the underlying biology than static approaches.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="programming.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reproducable-research.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"],
"instapaper": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
