<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Applied Population Genetics</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A textbook for the use of R in spatial genetic data analysis.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Applied Population Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="media/TitlePage_small.png" />
  <meta property="og:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="github-repo" content="dyerlab/AMG" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Applied Population Genetics" />
  
  <meta name="twitter:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="twitter:image" content="media/TitlePage_small.png" />

<meta name="author" content="Rodney J. Dyer">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="pedigrees.html">
<link rel="next" href="mapping-populations.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-0.8/htmlwidgets.js"></script>
<link href="libs/leaflet-0.7.7/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-0.7.7/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<link href="libs/leaflet-label-0.2.2/leaflet.label.css" rel="stylesheet" />
<script src="libs/leaflet-label-0.2.2/leaflet.label.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4-compressed.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4leaflet.js"></script>
<script src="libs/leaflet-binding-1.1.0/leaflet.js"></script>
<script src="libs/leaflet-providers-1.0.27/leaflet-providers.js"></script>
<script src="libs/leaflet-providers-plugin-1.1.0/leaflet-providers-plugin.js"></script>
<script src="libs/d3-4.5.0/d3.min.js"></script>
<script src="libs/sankey-1/sankey.js"></script>
<script src="libs/sankeyNetwork-binding-0.3/sankeyNetwork.js"></script>
<script src="libs/dendroNetwork-binding-0.3/dendroNetwork.js"></script>
<script src="libs/simpleNetwork-binding-0.3/simpleNetwork.js"></script>
<script src="libs/forceNetwork-binding-0.3/forceNetwork.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="media/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Applied Population Genetics</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#logistics"><i class="fa fa-check"></i>Logistics</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#typography"><i class="fa fa-check"></i>Typography</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#interactive-content"><i class="fa fa-check"></i>Interactive Content</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#dedication"><i class="fa fa-check"></i>Dedication</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html"><i class="fa fa-check"></i><b>1</b> The R Ecosystem</a><ul>
<li class="chapter" data-level="1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#getting-r-configured"><i class="fa fa-check"></i><b>1.1</b> Getting R Configured</a><ul>
<li class="chapter" data-level="1.1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
<li class="chapter" data-level="1.1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#cran"><i class="fa fa-check"></i><b>1.1.2</b> CRAN</a></li>
<li class="chapter" data-level="1.1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#github"><i class="fa fa-check"></i><b>1.1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.1.4" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#bioconductor"><i class="fa fa-check"></i><b>1.1.4</b> Bioconductor</a></li>
<li class="chapter" data-level="1.1.5" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#troublesome-packages"><i class="fa fa-check"></i><b>1.1.5</b> Troublesome Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#libraries-used-in-text"><i class="fa fa-check"></i><b>1.2</b> Libraries Used in Text</a></li>
<li class="chapter" data-level="1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#the-rstudio-environment"><i class="fa fa-check"></i><b>1.3</b> The RStudio Environment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>2</b> Data Types</a><ul>
<li class="chapter" data-level="2.1" data-path="data-types.html"><a href="data-types.html#numeric-data-types"><i class="fa fa-check"></i><b>2.1</b> Numeric Data Types</a><ul>
<li class="chapter" data-level="2.1.1" data-path="data-types.html"><a href="data-types.html#coercion-to-numeric"><i class="fa fa-check"></i><b>2.1.1</b> Coercion to Numeric</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-types.html"><a href="data-types.html#characters"><i class="fa fa-check"></i><b>2.2</b> Characters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="data-types.html"><a href="data-types.html#concatenation-of-characters"><i class="fa fa-check"></i><b>2.2.1</b> Concatenation of Characters</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types.html"><a href="data-types.html#coercion-to-characters"><i class="fa fa-check"></i><b>2.2.2</b> Coercion to Characters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types.html"><a href="data-types.html#factors"><i class="fa fa-check"></i><b>2.3</b> Factors</a><ul>
<li class="chapter" data-level="2.3.1" data-path="data-types.html"><a href="data-types.html#missing-levels-in-factors"><i class="fa fa-check"></i><b>2.3.1</b> Missing Levels in Factors</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-types.html"><a href="data-types.html#logical-types"><i class="fa fa-check"></i><b>2.4</b> Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-containers.html"><a href="data-containers.html"><i class="fa fa-check"></i><b>3</b> Data Containers</a><ul>
<li class="chapter" data-level="3.1" data-path="data-containers.html"><a href="data-containers.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a></li>
<li class="chapter" data-level="3.2" data-path="data-containers.html"><a href="data-containers.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="data-containers.html"><a href="data-containers.html#lists"><i class="fa fa-check"></i><b>3.3</b> Lists</a></li>
<li class="chapter" data-level="3.4" data-path="data-containers.html"><a href="data-containers.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a><ul>
<li class="chapter" data-level="3.4.1" data-path="data-containers.html"><a href="data-containers.html#indexing-data-frames"><i class="fa fa-check"></i><b>3.4.1</b> Indexing Data Frames</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="programming.html"><a href="programming.html"><i class="fa fa-check"></i><b>4</b> Programming</a><ul>
<li class="chapter" data-level="4.1" data-path="programming.html"><a href="programming.html#function-writing"><i class="fa fa-check"></i><b>4.1</b> Function Writing</a></li>
<li class="chapter" data-level="4.2" data-path="programming.html"><a href="programming.html#variable-scope"><i class="fa fa-check"></i><b>4.2</b> Variable Scope</a></li>
<li class="chapter" data-level="4.3" data-path="programming.html"><a href="programming.html#decision-making"><i class="fa fa-check"></i><b>4.3</b> Decision Making</a><ul>
<li class="chapter" data-level="4.3.1" data-path="programming.html"><a href="programming.html#the-if-pattern"><i class="fa fa-check"></i><b>4.3.1</b> The if Pattern</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="programming.html"><a href="programming.html#the-if-else-pattern"><i class="fa fa-check"></i><b>4.4</b> The <code>if-else</code> Pattern</a></li>
<li class="chapter" data-level="4.5" data-path="programming.html"><a href="programming.html#flow-control"><i class="fa fa-check"></i><b>4.5</b> Flow Control</a><ul>
<li class="chapter" data-level="4.5.1" data-path="programming.html"><a href="programming.html#the-for-loop"><i class="fa fa-check"></i><b>4.5.1</b> The for() Loop</a></li>
<li class="chapter" data-level="4.5.2" data-path="programming.html"><a href="programming.html#short-circuiting-the-loop"><i class="fa fa-check"></i><b>4.5.2</b> Short Circuiting the Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>5</b> Graphics</a><ul>
<li class="chapter" data-level="5.1" data-path="graphics.html"><a href="graphics.html#univariate-data"><i class="fa fa-check"></i><b>5.1</b> Univariate Data</a></li>
<li class="chapter" data-level="5.2" data-path="graphics.html"><a href="graphics.html#plotting-parameters"><i class="fa fa-check"></i><b>5.2</b> Plotting Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="graphics.html"><a href="graphics.html#density-plots"><i class="fa fa-check"></i><b>5.3</b> Density Plots</a></li>
<li class="chapter" data-level="5.4" data-path="graphics.html"><a href="graphics.html#overlaying-plots"><i class="fa fa-check"></i><b>5.4</b> Overlaying Plots</a></li>
<li class="chapter" data-level="5.5" data-path="graphics.html"><a href="graphics.html#boxplots"><i class="fa fa-check"></i><b>5.5</b> Boxplots</a></li>
<li class="chapter" data-level="5.6" data-path="graphics.html"><a href="graphics.html#scatter-plots"><i class="fa fa-check"></i><b>5.6</b> Scatter Plots</a></li>
<li class="chapter" data-level="5.7" data-path="graphics.html"><a href="graphics.html#multiple-plots"><i class="fa fa-check"></i><b>5.7</b> Multiple Plots</a></li>
<li class="chapter" data-level="5.8" data-path="graphics.html"><a href="graphics.html#color-palettes"><i class="fa fa-check"></i><b>5.8</b> Color Palettes</a></li>
<li class="chapter" data-level="5.9" data-path="graphics.html"><a href="graphics.html#rcolorbrewer"><i class="fa fa-check"></i><b>5.9</b> RColorBrewer</a></li>
<li class="chapter" data-level="5.10" data-path="graphics.html"><a href="graphics.html#saving-imagery"><i class="fa fa-check"></i><b>5.10</b> Saving Imagery</a></li>
<li class="chapter" data-level="5.11" data-path="graphics.html"><a href="graphics.html#interactive-graphics"><i class="fa fa-check"></i><b>5.11</b> Interactive Graphics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="reproducable-research.html"><a href="reproducable-research.html"><i class="fa fa-check"></i><b>6</b> Reproducable Research</a><ul>
<li class="chapter" data-level="6.1" data-path="reproducable-research.html"><a href="reproducable-research.html#keeping-data-on-google-drive"><i class="fa fa-check"></i><b>6.1</b> Keeping Data on Google Drive</a></li>
<li class="chapter" data-level="6.2" data-path="reproducable-research.html"><a href="reproducable-research.html#integrating-into-slack"><i class="fa fa-check"></i><b>6.2</b> Integrating into Slack</a><ul>
<li class="chapter" data-level="6.2.1" data-path="reproducable-research.html"><a href="reproducable-research.html#fixing-problem-with-version-1.4.1"><i class="fa fa-check"></i><b>6.2.1</b> Fixing Problem with Version 1.4.1</a></li>
<li class="chapter" data-level="6.2.2" data-path="reproducable-research.html"><a href="reproducable-research.html#configuration"><i class="fa fa-check"></i><b>6.2.2</b> Configuration</a></li>
<li class="chapter" data-level="6.2.3" data-path="reproducable-research.html"><a href="reproducable-research.html#sending-text-and-graphical-ouput"><i class="fa fa-check"></i><b>6.2.3</b> Sending Text and Graphical Ouput</a></li>
<li class="chapter" data-level="6.2.4" data-path="reproducable-research.html"><a href="reproducable-research.html#markdown"><i class="fa fa-check"></i><b>6.2.4</b> Markdown</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>I Single Population</b></span></li>
<li class="chapter" data-level="7" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html"><i class="fa fa-check"></i><b>7</b> Hardy-Weinberg Equilibrium</a><ul>
<li class="chapter" data-level="7.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#genetic-assumptions"><i class="fa fa-check"></i><b>7.1</b> Genetic Assumptions</a><ul>
<li class="chapter" data-level="7.1.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#punnet-squares"><i class="fa fa-check"></i><b>7.1.1</b> Punnet Squares</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#evolutionary-assumptions"><i class="fa fa-check"></i><b>7.2</b> Evolutionary Assumptions</a></li>
<li class="chapter" data-level="7.3" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#the-mechanics"><i class="fa fa-check"></i><b>7.3</b> The Mechanics</a></li>
<li class="chapter" data-level="7.4" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#trans-generational-transition-probabilities"><i class="fa fa-check"></i><b>7.4</b> Trans-Generational Transition Probabilities</a></li>
<li class="chapter" data-level="7.5" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#consequences-of-hwe"><i class="fa fa-check"></i><b>7.5</b> Consequences of HWE</a></li>
<li class="chapter" data-level="7.6" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#zygotic-gene-flow"><i class="fa fa-check"></i><b>7.6</b> Zygotic Gene Flow</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="genetic-drift.html"><a href="genetic-drift.html"><i class="fa fa-check"></i><b>8</b> Genetic Drift</a><ul>
<li class="chapter" data-level="8.1" data-path="genetic-drift.html"><a href="genetic-drift.html#consequences-of-genetic-drift"><i class="fa fa-check"></i><b>8.1</b> Consequences of Genetic Drift</a><ul>
<li class="chapter" data-level="8.1.1" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-fixation"><i class="fa fa-check"></i><b>8.1.1</b> Time to Fixation</a></li>
<li class="chapter" data-level="8.1.2" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-allele-loss"><i class="fa fa-check"></i><b>8.1.2</b> Time to Allele Loss</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="genetic-drift.html"><a href="genetic-drift.html#mutation-drift"><i class="fa fa-check"></i><b>8.2</b> Mutation &amp; Drift</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="effective-population-size.html"><a href="effective-population-size.html"><i class="fa fa-check"></i><b>9</b> Effective Population Size</a><ul>
<li class="chapter" data-level="9.1" data-path="effective-population-size.html"><a href="effective-population-size.html#demographic-effective-population-sizes"><i class="fa fa-check"></i><b>9.1</b> Demographic Effective Population Sizes</a></li>
<li class="chapter" data-level="9.2" data-path="effective-population-size.html"><a href="effective-population-size.html#genetic-effective-population-size"><i class="fa fa-check"></i><b>9.2</b> Genetic Effective Population Size</a><ul>
<li class="chapter" data-level="9.2.1" data-path="effective-population-size.html"><a href="effective-population-size.html#inbreeding-effective-population-size"><i class="fa fa-check"></i><b>9.2.1</b> Inbreeding Effective Population Size</a></li>
<li class="chapter" data-level="9.2.2" data-path="effective-population-size.html"><a href="effective-population-size.html#variance-effective-population-size"><i class="fa fa-check"></i><b>9.2.2</b> Variance Effective Population Size</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="effective-population-size.html"><a href="effective-population-size.html#fluctuating-population-size"><i class="fa fa-check"></i><b>9.3</b> Fluctuating Population Size</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutation.html"><a href="mutation.html"><i class="fa fa-check"></i><b>10</b> Mutation</a><ul>
<li class="chapter" data-level="10.1" data-path="mutation.html"><a href="mutation.html#mutation-models"><i class="fa fa-check"></i><b>10.1</b> Mutation Models</a><ul>
<li class="chapter" data-level="10.1.1" data-path="mutation.html"><a href="mutation.html#fixed-allele-model"><i class="fa fa-check"></i><b>10.1.1</b> Fixed Allele Model</a></li>
<li class="chapter" data-level="10.1.2" data-path="mutation.html"><a href="mutation.html#infinite-allele-model"><i class="fa fa-check"></i><b>10.1.2</b> Infinite Allele Model</a></li>
<li class="chapter" data-level="10.1.3" data-path="mutation.html"><a href="mutation.html#stepwise-mutation-model"><i class="fa fa-check"></i><b>10.1.3</b> Stepwise Mutation Model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="mutation.html"><a href="mutation.html#mutation-and-inbreeding"><i class="fa fa-check"></i><b>10.2</b> Mutation and Inbreeding</a></li>
<li class="chapter" data-level="10.3" data-path="mutation.html"><a href="mutation.html#estimating-mutation"><i class="fa fa-check"></i><b>10.3</b> Estimating Mutation</a><ul>
<li class="chapter" data-level="10.3.1" data-path="mutation.html"><a href="mutation.html#phenotypes"><i class="fa fa-check"></i><b>10.3.1</b> Phenotypes</a></li>
<li class="chapter" data-level="10.3.2" data-path="mutation.html"><a href="mutation.html#genotypes"><i class="fa fa-check"></i><b>10.3.2</b> Genotypes</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Nonrandom Rating</b></span></li>
<li class="chapter" data-level="11" data-path="selfing.html"><a href="selfing.html"><i class="fa fa-check"></i><b>11</b> Selfing</a><ul>
<li class="chapter" data-level="11.1" data-path="selfing.html"><a href="selfing.html#changes-in-f"><i class="fa fa-check"></i><b>11.1</b> Changes in F</a><ul>
<li class="chapter" data-level="11.1.1" data-path="selfing.html"><a href="selfing.html#theoretical-expectations"><i class="fa fa-check"></i><b>11.1.1</b> Theoretical Expectations</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="selfing.html"><a href="selfing.html#simulation-example"><i class="fa fa-check"></i><b>11.2</b> Simulation Example</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html"><i class="fa fa-check"></i><b>12</b> Mixed Mating Systems</a><ul>
<li class="chapter" data-level="12.1" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#mixed-mating-genotype-frequency-equilibriums"><i class="fa fa-check"></i><b>12.1</b> Mixed Mating Genotype Frequency Equilibriums</a></li>
<li class="chapter" data-level="12.2" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#estimating-selfing-rates"><i class="fa fa-check"></i><b>12.2</b> Estimating Selfing Rates</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wahlund-effects.html"><a href="wahlund-effects.html"><i class="fa fa-check"></i><b>13</b> Wahlund Effects</a></li>
<li class="part"><span><b>III Parent Offspring Data</b></span></li>
<li class="chapter" data-level="14" data-path="pedigrees.html"><a href="pedigrees.html"><i class="fa fa-check"></i><b>14</b> Pedigrees</a><ul>
<li class="chapter" data-level="14.1" data-path="pedigrees.html"><a href="pedigrees.html#expectations-for-f-in-a-pedigree."><i class="fa fa-check"></i><b>14.1</b> Expectations for F in a pedigree.</a></li>
<li class="chapter" data-level="14.2" data-path="pedigrees.html"><a href="pedigrees.html#drawing-pedigrees"><i class="fa fa-check"></i><b>14.2</b> Drawing Pedigrees</a></li>
<li class="chapter" data-level="14.3" data-path="pedigrees.html"><a href="pedigrees.html#skills"><i class="fa fa-check"></i><b>14.3</b> Skills</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="parentage-analyses.html"><a href="parentage-analyses.html"><i class="fa fa-check"></i><b>15</b> Parentage Analyses</a><ul>
<li class="chapter" data-level="15.1" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-vs.maternity"><i class="fa fa-check"></i><b>15.1</b> Paternity vs. Maternity</a></li>
<li class="chapter" data-level="15.2" data-path="parentage-analyses.html"><a href="parentage-analyses.html#probability-of-exclusion"><i class="fa fa-check"></i><b>15.2</b> Probability of Exclusion</a></li>
<li class="chapter" data-level="15.3" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-exclusion"><i class="fa fa-check"></i><b>15.3</b> Paternity Exclusion</a></li>
<li class="chapter" data-level="15.4" data-path="parentage-analyses.html"><a href="parentage-analyses.html#fractional-paternity"><i class="fa fa-check"></i><b>15.4</b> Fractional Paternity</a></li>
<li class="chapter" data-level="15.5" data-path="parentage-analyses.html"><a href="parentage-analyses.html#dispersal-kernels"><i class="fa fa-check"></i><b>15.5</b> Dispersal Kernels</a></li>
</ul></li>
<li class="part"><span><b>IV Subdivided Populations</b></span></li>
<li class="chapter" data-level="16" data-path="mapping-populations.html"><a href="mapping-populations.html"><i class="fa fa-check"></i><b>16</b> Mapping Populations</a><ul>
<li class="chapter" data-level="16.1" data-path="mapping-populations.html"><a href="mapping-populations.html#procedure"><i class="fa fa-check"></i><b>16.1</b> Procedure</a></li>
<li class="chapter" data-level="16.2" data-path="mapping-populations.html"><a href="mapping-populations.html#plotting-population-locations"><i class="fa fa-check"></i><b>16.2</b> Plotting Population Locations</a></li>
<li class="chapter" data-level="16.3" data-path="mapping-populations.html"><a href="mapping-populations.html#population-allele-frequencies"><i class="fa fa-check"></i><b>16.3</b> Population Allele Frequencies</a></li>
<li class="chapter" data-level="16.4" data-path="mapping-populations.html"><a href="mapping-populations.html#dynamic-maps"><i class="fa fa-check"></i><b>16.4</b> Dynamic maps</a><ul>
<li class="chapter" data-level="16.4.1" data-path="mapping-populations.html"><a href="mapping-populations.html#sankey-networks"><i class="fa fa-check"></i><b>16.4.1</b> Sankey Networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="population-models.html"><a href="population-models.html"><i class="fa fa-check"></i><b>17</b> Population Models</a><ul>
<li class="chapter" data-level="17.1" data-path="population-models.html"><a href="population-models.html#models-of-partition"><i class="fa fa-check"></i><b>17.1</b> Models of Partition</a></li>
<li class="chapter" data-level="17.2" data-path="population-models.html"><a href="population-models.html#island-mainland-model"><i class="fa fa-check"></i><b>17.2</b> Island Mainland Model</a></li>
<li class="chapter" data-level="17.3" data-path="population-models.html"><a href="population-models.html#the-island-model"><i class="fa fa-check"></i><b>17.3</b> The Island Model</a></li>
<li class="chapter" data-level="17.4" data-path="population-models.html"><a href="population-models.html#stepping-stone-models"><i class="fa fa-check"></i><b>17.4</b> Stepping Stone Models</a></li>
<li class="chapter" data-level="17.5" data-path="population-models.html"><a href="population-models.html#the-general-model"><i class="fa fa-check"></i><b>17.5</b> The General Model</a></li>
<li class="chapter" data-level="17.6" data-path="population-models.html"><a href="population-models.html#general-strategies"><i class="fa fa-check"></i><b>17.6</b> General Strategies</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>18</b> Genetic Diversity</a><ul>
<li class="chapter" data-level="18.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#allelic-diversity"><i class="fa fa-check"></i><b>18.1</b> Allelic Diversity</a></li>
<li class="chapter" data-level="18.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#genotypic-diversity"><i class="fa fa-check"></i><b>18.2</b> Genotypic Diversity</a></li>
<li class="chapter" data-level="18.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#samples-from-several-locales"><i class="fa fa-check"></i><b>18.3</b> Samples from Several Locales</a></li>
<li class="chapter" data-level="18.4" data-path="genetic-diversity.html"><a href="genetic-diversity.html#multilocus-diversity"><i class="fa fa-check"></i><b>18.4</b> Multilocus Diversity</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="rarefaction.html"><a href="rarefaction.html"><i class="fa fa-check"></i><b>19</b> Rarefaction</a><ul>
<li class="chapter" data-level="19.1" data-path="rarefaction.html"><a href="rarefaction.html#mapping-diversity"><i class="fa fa-check"></i><b>19.1</b> Mapping Diversity</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="genetic-distances.html"><a href="genetic-distances.html"><i class="fa fa-check"></i><b>20</b> Genetic Distances</a><ul>
<li class="chapter" data-level="20.1" data-path="genetic-distances.html"><a href="genetic-distances.html#individual-genetic-distance"><i class="fa fa-check"></i><b>20.1</b> Individual Genetic Distance</a><ul>
<li class="chapter" data-level="20.1.1" data-path="genetic-distances.html"><a href="genetic-distances.html#euclidean-distance"><i class="fa fa-check"></i><b>20.1.1</b> Euclidean Distance</a></li>
<li class="chapter" data-level="20.1.2" data-path="genetic-distances.html"><a href="genetic-distances.html#amova-distance"><i class="fa fa-check"></i><b>20.1.2</b> AMOVA Distance</a></li>
<li class="chapter" data-level="20.1.3" data-path="genetic-distances.html"><a href="genetic-distances.html#bray-curtis-individual-distance"><i class="fa fa-check"></i><b>20.1.3</b> Bray-Curtis Individual Distance</a></li>
<li class="chapter" data-level="20.1.4" data-path="genetic-distances.html"><a href="genetic-distances.html#ladder-distance"><i class="fa fa-check"></i><b>20.1.4</b> Ladder Distance</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="genetic-distances.html"><a href="genetic-distances.html#population-level-genetic-distances"><i class="fa fa-check"></i><b>20.2</b> Population-Level Genetic Distances</a><ul>
<li class="chapter" data-level="20.2.1" data-path="genetic-distances.html"><a href="genetic-distances.html#czekanowski-manhattan-distance"><i class="fa fa-check"></i><b>20.2.1</b> Czekanowski (Manhattan) Distance</a></li>
<li class="chapter" data-level="20.2.2" data-path="genetic-distances.html"><a href="genetic-distances.html#rogers-euclidean-distance"><i class="fa fa-check"></i><b>20.2.2</b> Rogers (Euclidean) Distance</a></li>
<li class="chapter" data-level="20.2.3" data-path="genetic-distances.html"><a href="genetic-distances.html#neis-genetic-distance"><i class="fa fa-check"></i><b>20.2.3</b> Nei’s Genetic Distance</a></li>
<li class="chapter" data-level="20.2.4" data-path="genetic-distances.html"><a href="genetic-distances.html#conditional-genetic-distance"><i class="fa fa-check"></i><b>20.2.4</b> Conditional Genetic Distance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="genetic-structure.html"><a href="genetic-structure.html"><i class="fa fa-check"></i><b>21</b> Genetic Structure</a><ul>
<li class="chapter" data-level="21.1" data-path="genetic-structure.html"><a href="genetic-structure.html#f-statistics-and-thier-ilk"><i class="fa fa-check"></i><b>21.1</b> F-Statistics and Thier Ilk</a><ul>
<li class="chapter" data-level="21.1.1" data-path="genetic-structure.html"><a href="genetic-structure.html#issues-with-fixation-indices"><i class="fa fa-check"></i><b>21.1.1</b> Issues with Fixation Indices</a></li>
<li class="chapter" data-level="21.1.2" data-path="genetic-structure.html"><a href="genetic-structure.html#additional-x_st-like-parameters"><i class="fa fa-check"></i><b>21.1.2</b> Additional <span class="math inline">\(X_{ST}\)</span>-like Parameters</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="genetic-structure.html"><a href="genetic-structure.html#statistical-structure"><i class="fa fa-check"></i><b>21.2</b> Statistical Structure</a></li>
<li class="chapter" data-level="21.3" data-path="genetic-structure.html"><a href="genetic-structure.html#final-thoughts-on-structure"><i class="fa fa-check"></i><b>21.3</b> Final Thoughts on Structure</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="population-assignment.html"><a href="population-assignment.html"><i class="fa fa-check"></i><b>22</b> Population Assignment</a></li>
<li class="chapter" data-level="23" data-path="admixture.html"><a href="admixture.html"><i class="fa fa-check"></i><b>23</b> Admixture</a></li>
<li class="part"><span><b>V Statistical Models</b></span></li>
<li class="chapter" data-level="24" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html"><i class="fa fa-check"></i><b>24</b> Ordination of Genetic Data</a><ul>
<li class="chapter" data-level="24.1" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#hierarchical-clustering"><i class="fa fa-check"></i><b>24.1</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="24.2" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#interactive-tree-display"><i class="fa fa-check"></i><b>24.2</b> Interactive Tree Display</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="distance-matrices.html"><a href="distance-matrices.html"><i class="fa fa-check"></i><b>25</b> Distance Matrices</a><ul>
<li class="chapter" data-level="25.1" data-path="distance-matrices.html"><a href="distance-matrices.html#visual-comparisons"><i class="fa fa-check"></i><b>25.1</b> Visual Comparisons</a></li>
<li class="chapter" data-level="25.2" data-path="distance-matrices.html"><a href="distance-matrices.html#the-mantel-test"><i class="fa fa-check"></i><b>25.2</b> The Mantel Test</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="linear-models.html"><a href="linear-models.html"><i class="fa fa-check"></i><b>26</b> Linear Models</a><ul>
<li class="chapter" data-level="26.1" data-path="linear-models.html"><a href="linear-models.html#dependent-independent-relationships"><i class="fa fa-check"></i><b>26.1</b> Dependent &amp; Independent Relationships</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="distance-networks.html"><a href="distance-networks.html"><i class="fa fa-check"></i><b>27</b> Distance Networks</a><ul>
<li class="chapter" data-level="27.1" data-path="distance-networks.html"><a href="distance-networks.html#graph-structure"><i class="fa fa-check"></i><b>27.1</b> Graph Structure</a><ul>
<li class="chapter" data-level="27.1.1" data-path="distance-networks.html"><a href="distance-networks.html#node-edge-attributes"><i class="fa fa-check"></i><b>27.1.1</b> Node &amp; Edge Attributes</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="distance-networks.html"><a href="distance-networks.html#plotting-a-graphs"><i class="fa fa-check"></i><b>27.2</b> Plotting a Graphs</a></li>
<li class="chapter" data-level="27.3" data-path="distance-networks.html"><a href="distance-networks.html#mapping-networks"><i class="fa fa-check"></i><b>27.3</b> Mapping Networks</a></li>
<li class="chapter" data-level="27.4" data-path="distance-networks.html"><a href="distance-networks.html#genetic-distance-graphs"><i class="fa fa-check"></i><b>27.4</b> Genetic Distance Graphs</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="population-graphs.html"><a href="population-graphs.html"><i class="fa fa-check"></i><b>28</b> Population Graphs</a><ul>
<li class="chapter" data-level="28.1" data-path="population-graphs.html"><a href="population-graphs.html#adding-data-to-a-graph"><i class="fa fa-check"></i><b>28.1</b> Adding data to a graph</a><ul>
<li class="chapter" data-level="28.1.1" data-path="population-graphs.html"><a href="population-graphs.html#plotting-a-graph-using-ggplot2-routines"><i class="fa fa-check"></i><b>28.1.1</b> Plotting a graph using <strong>ggplot2</strong> routines</a></li>
<li class="chapter" data-level="28.1.2" data-path="population-graphs.html"><a href="population-graphs.html#reading-existing-popgraph-files"><i class="fa fa-check"></i><b>28.1.2</b> Reading Existing <strong>popgraph</strong> Files</a></li>
<li class="chapter" data-level="28.1.3" data-path="population-graphs.html"><a href="population-graphs.html#saving-population-graph-objects"><i class="fa fa-check"></i><b>28.1.3</b> Saving Population Graph Objects</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="population-graphs.html"><a href="population-graphs.html#interactive-networks"><i class="fa fa-check"></i><b>28.2</b> Interactive Networks</a></li>
<li class="chapter" data-level="28.3" data-path="population-graphs.html"><a href="population-graphs.html#spatial-population-graphs"><i class="fa fa-check"></i><b>28.3</b> Spatial Population Graphs</a><ul>
<li class="chapter" data-level="28.3.1" data-path="population-graphs.html"><a href="population-graphs.html#integrating-google-and-ggplot2-for-plotting"><i class="fa fa-check"></i><b>28.3.1</b> Integrating Google and <strong>ggplot2</strong> for Plotting</a></li>
<li class="chapter" data-level="28.3.2" data-path="population-graphs.html"><a href="population-graphs.html#integrating-raster-maps"><i class="fa fa-check"></i><b>28.3.2</b> Integrating Raster Maps</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="population-graphs.html"><a href="population-graphs.html#extracting-spatial-data-using-population-graphs"><i class="fa fa-check"></i><b>28.4</b> Extracting Spatial Data Using Population Graphs</a><ul>
<li class="chapter" data-level="28.4.1" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-data"><i class="fa fa-check"></i><b>28.4.1</b> Node Specific Data</a></li>
<li class="chapter" data-level="28.4.2" data-path="population-graphs.html"><a href="population-graphs.html#extracting-data-along-popgraph-edges"><i class="fa fa-check"></i><b>28.4.2</b> Extracting Data Along Popgraph Edges</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="population-graphs.html"><a href="population-graphs.html#extracting-graph-theoretic-parameters"><i class="fa fa-check"></i><b>28.5</b> Extracting Graph-Theoretic Parameters</a><ul>
<li class="chapter" data-level="28.5.1" data-path="population-graphs.html"><a href="population-graphs.html#matrix-representations-of-population-graph-objects"><i class="fa fa-check"></i><b>28.5.1</b> Matrix Representations of Population Graph Objects</a></li>
<li class="chapter" data-level="28.5.2" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-parameters"><i class="fa fa-check"></i><b>28.5.2</b> Node Specific Parameters</a></li>
<li class="chapter" data-level="28.5.3" data-path="population-graphs.html"><a href="population-graphs.html#edge-specific-parameters"><i class="fa fa-check"></i><b>28.5.3</b> Edge Specific Parameters</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="population-graphs.html"><a href="population-graphs.html#testing-for-topological-congruence"><i class="fa fa-check"></i><b>28.6</b> Testing for Topological Congruence</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="interactive-visualization.html"><a href="interactive-visualization.html"><i class="fa fa-check"></i><b>29</b> Interactive Visualization</a><ul>
<li class="chapter" data-level="29.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#procedure-1"><i class="fa fa-check"></i><b>29.1</b> Procedure</a><ul>
<li class="chapter" data-level="29.1.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#interactive-networks-1"><i class="fa fa-check"></i><b>29.1.1</b> Interactive Networks</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>VI Spatial Data</b></span></li>
<li class="chapter" data-level="30" data-path="vector-data.html"><a href="vector-data.html"><i class="fa fa-check"></i><b>30</b> Vector Data</a><ul>
<li class="chapter" data-level="30.1" data-path="vector-data.html"><a href="vector-data.html#points"><i class="fa fa-check"></i><b>30.1</b> Points</a></li>
<li class="chapter" data-level="30.2" data-path="vector-data.html"><a href="vector-data.html#lines"><i class="fa fa-check"></i><b>30.2</b> Lines</a></li>
<li class="chapter" data-level="30.3" data-path="vector-data.html"><a href="vector-data.html#polygons"><i class="fa fa-check"></i><b>30.3</b> Polygons</a></li>
<li class="chapter" data-level="30.4" data-path="vector-data.html"><a href="vector-data.html#saving-vector-objects"><i class="fa fa-check"></i><b>30.4</b> Saving Vector Objects</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="raster-data.html"><a href="raster-data.html"><i class="fa fa-check"></i><b>31</b> Raster Data</a><ul>
<li class="chapter" data-level="31.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters"><i class="fa fa-check"></i><b>31.1</b> Cropping Rasters</a><ul>
<li class="chapter" data-level="31.1.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-polygons"><i class="fa fa-check"></i><b>31.1.1</b> Cropping Rasters Via Polygons</a></li>
<li class="chapter" data-level="31.1.2" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-convex-hull"><i class="fa fa-check"></i><b>31.1.2</b> Cropping Rasters Via Convex Hull</a></li>
</ul></li>
<li class="chapter" data-level="31.2" data-path="raster-data.html"><a href="raster-data.html#modifying-rasters"><i class="fa fa-check"></i><b>31.2</b> Modifying Rasters</a></li>
<li class="chapter" data-level="31.3" data-path="raster-data.html"><a href="raster-data.html#extracting-point-data-from-rasters"><i class="fa fa-check"></i><b>31.3</b> Extracting Point Data From Rasters</a><ul>
<li class="chapter" data-level="31.3.1" data-path="raster-data.html"><a href="raster-data.html#stacks-of-rasters"><i class="fa fa-check"></i><b>31.3.1</b> Stacks of Rasters</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="raster-data.html"><a href="raster-data.html#rasters-ggplot"><i class="fa fa-check"></i><b>31.4</b> Rasters &amp; ggplot</a></li>
<li class="chapter" data-level="31.5" data-path="raster-data.html"><a href="raster-data.html#d-visualization"><i class="fa fa-check"></i><b>31.5</b> 3D Visualization</a></li>
<li class="chapter" data-level="31.6" data-path="raster-data.html"><a href="raster-data.html#saving-exporting-rasters"><i class="fa fa-check"></i><b>31.6</b> Saving &amp; Exporting Rasters</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="map-projections.html"><a href="map-projections.html"><i class="fa fa-check"></i><b>32</b> Map Projections</a><ul>
<li class="chapter" data-level="32.1" data-path="map-projections.html"><a href="map-projections.html#projections"><i class="fa fa-check"></i><b>32.1</b> Projections</a><ul>
<li class="chapter" data-level="32.1.1" data-path="map-projections.html"><a href="map-projections.html#equatorial-projections"><i class="fa fa-check"></i><b>32.1.1</b> Equatorial Projections</a></li>
<li class="chapter" data-level="32.1.2" data-path="map-projections.html"><a href="map-projections.html#azimuth-projections"><i class="fa fa-check"></i><b>32.1.2</b> Azimuth Projections</a></li>
<li class="chapter" data-level="32.1.3" data-path="map-projections.html"><a href="map-projections.html#polar-conic-projections"><i class="fa fa-check"></i><b>32.1.3</b> Polar Conic Projections</a></li>
<li class="chapter" data-level="32.1.4" data-path="map-projections.html"><a href="map-projections.html#miscellaneous-projections"><i class="fa fa-check"></i><b>32.1.4</b> Miscellaneous Projections</a></li>
<li class="chapter" data-level="32.1.5" data-path="map-projections.html"><a href="map-projections.html#reprojecting-rasters"><i class="fa fa-check"></i><b>32.1.5</b> Reprojecting Rasters</a></li>
<li class="chapter" data-level="32.1.6" data-path="map-projections.html"><a href="map-projections.html#projecting-in-ggplot"><i class="fa fa-check"></i><b>32.1.6</b> Projecting In GGPlot</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="map-projections.html"><a href="map-projections.html#coordinate-systems"><i class="fa fa-check"></i><b>32.2</b> Coordinate Systems</a><ul>
<li class="chapter" data-level="32.2.1" data-path="map-projections.html"><a href="map-projections.html#changing-datum"><i class="fa fa-check"></i><b>32.2.1</b> Changing Datum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="ecological-distance.html"><a href="ecological-distance.html"><i class="fa fa-check"></i><b>33</b> Ecological Distance</a><ul>
<li class="chapter" data-level="33.1" data-path="ecological-distance.html"><a href="ecological-distance.html#the-concept-of-resistance"><i class="fa fa-check"></i><b>33.1</b> The Concept of Resistance</a><ul>
<li class="chapter" data-level="33.1.1" data-path="ecological-distance.html"><a href="ecological-distance.html#absolute-resistance"><i class="fa fa-check"></i><b>33.1.1</b> Absolute Resistance</a></li>
</ul></li>
<li class="chapter" data-level="33.2" data-path="ecological-distance.html"><a href="ecological-distance.html#relative-resistance"><i class="fa fa-check"></i><b>33.2</b> Relative Resistance</a></li>
<li class="chapter" data-level="33.3" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-separation"><i class="fa fa-check"></i><b>33.3</b> Estimating Separation</a><ul>
<li class="chapter" data-level="33.3.1" data-path="ecological-distance.html"><a href="ecological-distance.html#least-cost-distance"><i class="fa fa-check"></i><b>33.3.1</b> Least Cost Distance</a></li>
<li class="chapter" data-level="33.3.2" data-path="ecological-distance.html"><a href="ecological-distance.html#all-paths-circuit-distance"><i class="fa fa-check"></i><b>33.3.2</b> All Paths (Circuit) Distance</a></li>
<li class="chapter" data-level="33.3.3" data-path="ecological-distance.html"><a href="ecological-distance.html#path-corridor-constraints"><i class="fa fa-check"></i><b>33.3.3</b> Path &amp; Corridor Constraints</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-correlations-in-distance"><i class="fa fa-check"></i><b>33.4</b> Estimating Correlations in Distance</a><ul>
<li class="chapter" data-level="33.4.1" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-least-cost-path"><i class="fa fa-check"></i><b>33.4.1</b> Pairwise Least Cost Path</a></li>
<li class="chapter" data-level="33.4.2" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-all-pairs"><i class="fa fa-check"></i><b>33.4.2</b> Pairwise All Pairs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="niche-modeling.html"><a href="niche-modeling.html"><i class="fa fa-check"></i><b>34</b> Niche Modeling</a></li>
<li class="chapter" data-level="35" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html"><i class="fa fa-check"></i><b>35</b> Eigenvector Maps</a><ul>
<li class="chapter" data-level="35.1" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html#eigenvectors-eigenvalues"><i class="fa fa-check"></i><b>35.1</b> Eigenvectors &amp; Eigenvalues</a></li>
</ul></li>
<li class="part"><span><b>VII Selection</b></span></li>
<li class="chapter" data-level="36" data-path="classical-selection-models.html"><a href="classical-selection-models.html"><i class="fa fa-check"></i><b>36</b> Classical Selection Models</a><ul>
<li class="chapter" data-level="36.1" data-path="classical-selection-models.html"><a href="classical-selection-models.html#the-strength-of-selection"><i class="fa fa-check"></i><b>36.1</b> The Strength of Selection</a></li>
</ul></li>
<li class="part"><span><b>VIII Backmatter</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="37" data-path="included-data-sets.html"><a href="included-data-sets.html"><i class="fa fa-check"></i><b>37</b> Included Data Sets</a><ul>
<li class="chapter" data-level="37.1" data-path="included-data-sets.html"><a href="included-data-sets.html#the-sonoran-desert-bark-beetle-araptus-attenuata"><i class="fa fa-check"></i><b>37.1</b> The Sonoran desert bark beetle, <em>Araptus attenuata</em></a><ul>
<li class="chapter" data-level="37.1.1" data-path="included-data-sets.html"><a href="included-data-sets.html#nuclear-markers"><i class="fa fa-check"></i><b>37.1.1</b> Nuclear Markers</a></li>
<li class="chapter" data-level="37.1.2" data-path="included-data-sets.html"><a href="included-data-sets.html#methylation-markers"><i class="fa fa-check"></i><b>37.1.2</b> Methylation Markers</a></li>
</ul></li>
<li class="chapter" data-level="37.2" data-path="included-data-sets.html"><a href="included-data-sets.html#the-flowering-dogwood-cornus-florida"><i class="fa fa-check"></i><b>37.2</b> The Flowering Dogwood <em>Cornus florida</em></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-author.html"><a href="about-the-author.html"><i class="fa fa-check"></i>About the Author</a></li>
<li class="divider"></li>
<li><a href="http://dyerlab.bio.vcu.edu" target="blank">the Dyer Laboratory</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Population Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="parentage-analyses" class="section level1 imageChapter">
<h1><span class="header-section-number">15</span> Parentage Analyses</h1>
<div class="chapter_image">
<img src="chapter_images/ch_butterfly.jpg">
</div>
<p>The analysis of parental and offspring data</p>
<ol style="list-style-type: decimal">
<li>Parentage-type analyses.<br />
</li>
<li>Analyses based upon multiple paternity</li>
</ol>
<p>Parentage analyses are used in a broad range of studies:</p>
<ul>
<li>Identical vs fraternal twins<br />
</li>
<li>Agricultural crop line differentiation<br />
</li>
<li>Differentiate between livestock/dog/cat breeds<br />
</li>
<li>Pathogenic strain identification (e.g., Hep-C strains <em>a-e</em>)<br />
</li>
<li>Assigning parentage to individuals<br />
</li>
<li>Assigning individuals to populations<br />
</li>
<li>Identifying the source of unknown tissues (e.g., gettin’ perps)</li>
</ul>
<p><img src="media/paternity.png" width="390" /></p>
<p>A statistical approach for identifying the parent(s) of a particular individual. This requires:</p>
<ol style="list-style-type: decimal">
<li>A set of genetic markers that are bi-parentally inherited</li>
<li>Variation in these markers</li>
<li>Some assumptions about the <em>prior</em> probability of the union of parents.</li>
</ol>
<div id="paternity-vs.maternity" class="section level2">
<h2><span class="header-section-number">15.1</span> Paternity vs. Maternity</h2>
<p>For single parent parentage analysis, it is either paternity or maternity that is being established. Here we assume that the other parent is definitely the biological parent of the individual (e.g., <span class="math inline">\(P(prior)=1\)</span>). This can be because:</p>
<ol style="list-style-type: decimal">
<li>The offspring was collected from the identified parent</li>
<li>There is other evidence that points to the identified parent being the biological one.</li>
</ol>
<p>The goal then is to determine who the unidentified parent is with some level of statistical probability.</p>
<p>Terms Used in Parentage</p>
<p>The following terms are commonly used in parentage analyses:</p>
<p><strong>Extra-Pair Paternity</strong> - Fertilization resulting from copulation outside a recognized pair bond.</p>
<p><strong>Multiple Paternity</strong> - Offspring produced from mating events with different sets of individuals.</p>
<p><strong>Paternity/Maternity Exclusion</strong> - Excluding an individual based upon an in-congruence in observed genetic data.</p>
</div>
<div id="probability-of-exclusion" class="section level2">
<h2><span class="header-section-number">15.2</span> Probability of Exclusion</h2>
<table>
<thead>
<tr class="header">
<th>Mother</th>
<th>Offspring</th>
<th>Excluded Dads</th>
<th>Probability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(A_1A_1\;(p^2_1)\)</span></td>
<td><span class="math inline">\(A_1A_1\;(p_1)\)</span></td>
<td><span class="math inline">\(A_2A_2\;(p_2^2)\)</span></td>
<td><span class="math inline">\(p_1^3p_2^2\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_1A_1\;(p^2_1)\)</span></td>
<td><span class="math inline">\(A_1A_2\;(p_2)\)</span></td>
<td><span class="math inline">\(A_1A_1\;(p_1^2)\)</span></td>
<td><span class="math inline">\(p_1^4p_2\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_1A_1\;(p^2_1)\)</span></td>
<td><span class="math inline">\(A_2A_2\;(0)\)</span></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_1A_2\;(2p_1p_2)\)</span></td>
<td><span class="math inline">\(A_1A_1\;(\frac{p_1}{2})\)</span></td>
<td><span class="math inline">\(A_2A_2\;(p_2^2)\)</span></td>
<td><span class="math inline">\(p_1^2p_2^3\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_1A_2\;(2p_1p_2)\)</span></td>
<td><span class="math inline">\(A_1A_2\;(\frac{1}{2})\)</span></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_1A_2\;(2p_1p_2)\)</span></td>
<td><span class="math inline">\(A_2A_2\;(\frac{p_2}{2})\)</span></td>
<td><span class="math inline">\(A_1A_1\;(p_1^2)\)</span></td>
<td><span class="math inline">\(p_1^3p_2^2\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_2A_2\;(p^2_2)\)</span></td>
<td><span class="math inline">\(A_1A_1\;(0)\)</span></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_2A_2\;(p^2_2)\)</span></td>
<td><span class="math inline">\(A_1A_2\;(p_1)\)</span></td>
<td><span class="math inline">\(A_2A_2\;(p_2^2)\)</span></td>
<td><span class="math inline">\(p_1p_2^4\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_2A_2\;(p^2_2)\)</span></td>
<td><span class="math inline">\(A_2A_2\;(p_2)\)</span></td>
<td><span class="math inline">\(A_1A_1\;(p_1^2)\)</span></td>
<td><span class="math inline">\(p_1^2p_2^3\)</span></td>
</tr>
</tbody>
</table>
<p>Probability of Exclusion</p>
<p><span class="math display">\[
P_{exc} = p_1^3p_2^2 + p_1^4p_2 +p_1^2p_2^3 +p_1^3p_2^2 +p_1p_2^4 +p_1^2p_2^3 \\
\]</span></p>
<p>which when simplified down a bit becomes</p>
<p><span class="math display">\[
P_{exc} = p_1p_2(1-p_1p_2)
\]</span></p>
<p>Single Locus Paternity Exclusion</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">by=</span><span class="fl">0.02</span>)
q &lt;-<span class="st"> </span><span class="dv">1</span>-p
Pexcl &lt;-<span class="st"> </span>p*q*(<span class="dv">1</span>-p*q)
<span class="kw">plot</span>(Pexcl ~<span class="st"> </span>p, <span class="dt">xlab=</span><span class="st">&quot;Allele frequency, p&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Paternity Exclusion Probability&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/chunk-parentage-1.png" width="672" /></p>
<p>Multilocus Exclusion</p>
<p>Exclusion probabilities are multiplicative properties.</p>
<p><span class="math display">\[
P_{excl} = 1 - \prod_{i=1}^\ell(1-P_{excl,\;i})
\]</span></p>
<p>Example</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">rep</span>( <span class="fl">0.5</span>*<span class="fl">0.5</span>*(<span class="dv">1</span><span class="fl">-0.5</span>*<span class="fl">0.5</span>), <span class="dv">5</span> )
p</code></pre></div>
<pre><code>## [1] 0.1875 0.1875 0.1875 0.1875 0.1875</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ptot &lt;-<span class="st"> </span><span class="dv">1</span>
for( i in <span class="dv">1</span>:<span class="kw">length</span>(p))
  ptot &lt;-<span class="st"> </span>ptot *<span class="st"> </span>(<span class="dv">1</span>-p[i])
<span class="dv">1</span>-ptot</code></pre></div>
<pre><code>## [1] 0.6459074</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="25%" />
<col width="27%" />
<col width="29%" />
</colgroup>
<thead>
<tr class="header">
<th>Mother</th>
<th>Offspring</th>
<th>Excluded Father <span class="math inline">\(A_xA_y\)</span></th>
<th>Probability of Exclusion</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(A_iA_i\;(p_i^2)\)</span></td>
<td><span class="math inline">\(A_iA_i\;(p_i)\)</span></td>
<td><span class="math inline">\(x,y\ne i\;\; (1-p_i)^2\)</span></td>
<td><span class="math inline">\(p_i^3(1-p_i)^2\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_iA_i\;(p_i^2)\)</span></td>
<td><span class="math inline">\(A_iA_j\;(p_j)\)</span></td>
<td><span class="math inline">\(x,y\ne j\;\; (1-p_j)^2\)</span></td>
<td><span class="math inline">\(p_i^2p_j(1-p_i)^2\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_iA_j\;(2p_ip_j)\)</span></td>
<td><span class="math inline">\(A_iA_i\;(\frac{p_i}{2})\)</span></td>
<td><span class="math inline">\(x,y \ne i\;\; (1-p_i)^2\)</span></td>
<td><span class="math inline">\(p_i^2p_j(1-p_i)^2\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_iA_j\;(2p_ip_j)\)</span></td>
<td><span class="math inline">\(A_iA_j\;(\frac{p_i+p_j}{2})\)</span></td>
<td><span class="math inline">\(x,y \ne i,j\;\; (1-p_i-p_j)^2\)</span></td>
<td><span class="math inline">\(p_ip_j(p_i+p_j)(1-p_i-p_j)^2\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_iA_j\;(2p_ip_j)\)</span></td>
<td><span class="math inline">\(A_iA_k\;(\frac{p_k}{2})\)</span></td>
<td><span class="math inline">\(x,y \ne k\;\; (1-p_k)^2\)</span></td>
<td><span class="math inline">\(p_ip_jp_k(1-p_k)^2\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(A_iA_j\;(2p_ip_j)\)</span></td>
<td><span class="math inline">\(A_jA_k\;(\frac{p_k}{2})\)</span></td>
<td><span class="math inline">\(x,y \ne k\;\; (1-p_k)^2\)</span></td>
<td><span class="math inline">\(p_ip_jp_k(1-p_k)^2\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(A_iA_j\;(2p_ip_j)\)</span></td>
<td><span class="math inline">\(A_jA_j\;(\frac{p_j}{2})\)</span></td>
<td><span class="math inline">\(x,y \ne j\;\; (1-p_j)^2\)</span></td>
<td><span class="math inline">\(p_ip_j^2(1-p_j)^2\)</span></td>
</tr>
</tbody>
</table>
</div>
<div id="paternity-exclusion" class="section level2">
<h2><span class="header-section-number">15.3</span> Paternity Exclusion</h2>
<p>Likelihood Ratios</p>
<p>A likelihood ratio is given by:</p>
<p><span class="math display">\[
LR = \frac{H_P}{H_D}
\]</span></p>
<p>where the <span class="math inline">\(H_X\)</span> values are the hypotheses probabilities.</p>
<p>Nomenclature For Parentage</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Identifier</th>
<th>Genotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Female Parent</td>
<td><span class="math inline">\(FP_i\)</span></td>
<td><span class="math inline">\(\alpha_i\)</span></td>
</tr>
<tr class="even">
<td>Putative Male Parent</td>
<td><span class="math inline">\(MP_j\)</span></td>
<td><span class="math inline">\(\beta_j\)</span></td>
</tr>
<tr class="odd">
<td>Offspring</td>
<td><span class="math inline">\(O_k\)</span></td>
<td><span class="math inline">\(\gamma_k\)</span></td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(\;\)</span></p>
<p><strong>Paternal Probability</strong> The posterior odds of paternity versus non-paternity given the totality of genetic information.</p>
<p>Likelihood Ratios | Genetic Equivalences</p>
<p>The likelihood of one hypothesis, <span class="math inline">\(H_1\)</span> relative to another <span class="math inline">\(H_2\)</span> is:</p>
<p><span class="math display">\[
L(H_1,H_2|D) = \frac{P(D|H_1)}{P(D|H_2)}
\]</span></p>
<p>where</p>
<p><span class="math display">\[
P(D|H) = T(\gamma | \alpha, \beta)P(\alpha)P(\beta)
\]</span></p>
<p>Assuming <span class="math inline">\(H_1:\)</span> states that <span class="math inline">\(\beta\)</span> is the real father of <span class="math inline">\(\gamma\)</span> on <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(H_2:\)</span> states that he is just a random individual in the population is:</p>
<p><span class="math display">\[
L(H_1,H_2|\alpha,\beta,\gamma)  = \frac{P(D|\alpha,\beta,\gamma)}{P(D|\alpha,\gamma)}  
\]</span></p>
<p>which can be simplified to:</p>
<p><span class="math display">\[
\lambda_j = \frac{P(\alpha_i,\beta_j,\gamma_k|\mathrm{paternity})}{P(\alpha_i,\beta_j,\gamma_k|\mathrm{non-paternity})} \\ 
            = \frac{P(\beta_j)P(\alpha_i)T(\gamma_k|\alpha_i,\beta_j)}{P(\beta_j)P(\alpha_i)T(\gamma_k|\alpha_i)} \\ 
                  = \frac{T(\gamma_k|\alpha_i,\beta_j)}{T(\gamma_k|\alpha_i)}
\]</span></p>
<p>where <span class="math inline">\(T(X|Y)\)</span> is the Mendelian transition probability of offspring <span class="math inline">\(X\)</span> given parent <span class="math inline">\(Y\)</span>.</p>
<p>Assumptions in Model of Paternity Likelihood</p>
<p>The basic paternity exclusion model assumes:</p>
<ol style="list-style-type: decimal">
<li>Completely random mating (can be modified by changin <em>priors</em>)</li>
<li>Independent assortment of alleles</li>
</ol>
<p>Likelihood Example</p>
<p>Consider the maternal individual whose genotypes are: <span class="math display">\[
FP = \{AA,\;Bb,\;CC,\;Dd\}
\]</span></p>
<p>Whose <span class="math inline">\(i^{th}\)</span> offspring has the genotypes:</p>
<p><span class="math display">\[
O_i = \{AA,\;BB,\;Cc,\;dd\}
\]</span></p>
<p>Likelihood Example | <span class="math inline">\(T(O|FP)\)</span></p>
<p>The transition probability, <span class="math inline">\(T(O|FP)\)</span>, is then:</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Locus1</th>
<th>Locus2</th>
<th>Locus3</th>
<th>Locus4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(FP\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(Bb\)</span></td>
<td><span class="math inline">\(CC\)</span></td>
<td><span class="math inline">\(Dd\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(O_i\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(Cc\)</span></td>
<td><span class="math inline">\(dd\)</span></td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(T(O|FP) = 1*0.5*1*0.5 = 0.25\)</span></p>
<p>Likelihood Example | Putative Male Parents</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Locus1</th>
<th>Locus2</th>
<th>Locus3</th>
<th>Locus4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(MP_1\)</span></td>
<td><span class="math inline">\(Aa\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(cc\)</span></td>
<td><span class="math inline">\(Dd\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(MP_2\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(Cc\)</span></td>
<td><span class="math inline">\(dd\)</span></td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(\;\)</span></p>
<p>Which one of the potential fathers is the <font class="alert">most likely</font> parent?</p>
<p>Likelihood Example | First Putative Father</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Locus1</th>
<th>Locus2</th>
<th>Locus3</th>
<th>Locus4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(FP\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(Bb\)</span></td>
<td><span class="math inline">\(CC\)</span></td>
<td><span class="math inline">\(Dd\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(MP_1\)</span></td>
<td><span class="math inline">\(Aa\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(cc\)</span></td>
<td><span class="math inline">\(Dd\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(O_i\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(Cc\)</span></td>
<td><span class="math inline">\(dd\)</span></td>
</tr>
<tr class="even">
<td>$T(O</td>
<td>FP,MP)$</td>
<td>0.5</td>
<td>0.5</td>
<td>1.0</td>
</tr>
</tbody>
</table>
<p><span class="math display">\[
T(O_1|FP,MP_1) = 0.5 * 0.5 * 1.0 * 0.25 = 0.0625
\]</span></p>
<p>And</p>
<p><span class="math display">\[
\lambda_1 = \frac{T(O_i|FP,MP_1)}{T(O_i|FP)} = \frac{0.0625}{0.25} = 0.25
\]</span></p>
<p>Likelihood Example | Second Putative Father</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Locus1</th>
<th>Locus2</th>
<th>Locus3</th>
<th>Locus4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(FP\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(Bb\)</span></td>
<td><span class="math inline">\(CC\)</span></td>
<td><span class="math inline">\(Dd\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(MP_2\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(Cc\)</span></td>
<td><span class="math inline">\(dd\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(O_i\)</span></td>
<td><span class="math inline">\(AA\)</span></td>
<td><span class="math inline">\(BB\)</span></td>
<td><span class="math inline">\(Cc\)</span></td>
<td><span class="math inline">\(dd\)</span></td>
</tr>
<tr class="even">
<td>$T(O</td>
<td>FP,MP)$</td>
<td>1.0</td>
<td>0.5</td>
<td>0.5</td>
</tr>
</tbody>
</table>
<p><span class="math display">\[
T(O_1|FP,MP_2) = 0.5 * 0.5 * 1.0 * 0.25 = 0.125
\]</span></p>
<p>And</p>
<p><span class="math display">\[
\lambda_2 = \frac{T(O_i|FP,MP_1)}{T(O_i|FP)} = \frac{0.125}{0.25} = 0.5
\]</span></p>
<p>Likelihood Example | Interpretation of Results {.build}</p>
<p>Most likely parent is <span class="math inline">\(MP_2\)</span> because <span class="math inline">\(\lambda_2 = 0.5 &gt; \lambda_1 = 0.25\)</span>.</p>
<p><span class="math inline">\(\;\)</span></p>
<p>Does this mean that <span class="math inline">\(MP_2\)</span> is the <font class="alert">real</font> parent?</p>
<p>Likelihood Example | In Class Exercise - Whose the daddies?</p>
<table>
<thead>
<tr class="header">
<th>Individual</th>
<th>Locus 1</th>
<th>Locus 2</th>
<th>Locus 3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Mother</td>
<td><span class="math inline">\(A_1A_1\)</span></td>
<td><span class="math inline">\(B_1B_3\)</span></td>
<td><span class="math inline">\(C_1C_1\)</span></td>
</tr>
<tr class="even">
<td>Offspring 1</td>
<td><span class="math inline">\(A_1A_2\)</span></td>
<td><span class="math inline">\(B_1B_3\)</span></td>
<td><span class="math inline">\(C_1C_2\)</span></td>
</tr>
<tr class="odd">
<td>Offspring 2</td>
<td><span class="math inline">\(A_1A_1\)</span></td>
<td><span class="math inline">\(B_3B_3\)</span></td>
<td><span class="math inline">\(C_1C_1\)</span></td>
</tr>
<tr class="even">
<td>Offspring 3</td>
<td><span class="math inline">\(A_1A_1\)</span></td>
<td><span class="math inline">\(B_1B_1\)</span></td>
<td><span class="math inline">\(C_1C_1\)</span></td>
</tr>
<tr class="odd">
<td>Dad 1</td>
<td><span class="math inline">\(A_1A_2\)</span></td>
<td><span class="math inline">\(B_2B_3\)</span></td>
<td><span class="math inline">\(C_1C_1\)</span></td>
</tr>
<tr class="even">
<td>Dad 2</td>
<td><span class="math inline">\(A_2A_2\)</span></td>
<td><span class="math inline">\(B_1B_1\)</span></td>
<td><span class="math inline">\(C_1C_2\)</span></td>
</tr>
<tr class="odd">
<td>Dad 3</td>
<td><span class="math inline">\(A_1A_1\)</span></td>
<td><span class="math inline">\(B_2B_3\)</span></td>
<td><span class="math inline">\(C_1C_2\)</span></td>
</tr>
<tr class="even">
<td>Dad 4</td>
<td><span class="math inline">\(A_1A_1\)</span></td>
<td><span class="math inline">\(B_1B_1\)</span></td>
<td><span class="math inline">\(C_2C_2\)</span></td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(gstudio)
loci &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Locus-A&quot;</span>,<span class="st">&quot;Locus-B&quot;</span>,<span class="st">&quot;Locus-C&quot;</span>,<span class="st">&quot;Locus-D&quot;</span>)
freqs &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Locus =</span> <span class="kw">rep</span>(loci, <span class="dt">each =</span> <span class="dv">4</span>), 
                    <span class="dt">Allele =</span> <span class="kw">rep</span>(LETTERS[<span class="dv">1</span>:<span class="dv">4</span>], <span class="dt">times =</span> <span class="dv">4</span>), 
                    <span class="dt">Frequency =</span> <span class="fl">0.25</span>)
freqs</code></pre></div>
<pre><code>##      Locus Allele Frequency
## 1  Locus-A      A      0.25
## 2  Locus-A      B      0.25
## 3  Locus-A      C      0.25
## 4  Locus-A      D      0.25
## 5  Locus-B      A      0.25
## 6  Locus-B      B      0.25
## 7  Locus-B      C      0.25
## 8  Locus-B      D      0.25
## 9  Locus-C      A      0.25
## 10 Locus-C      B      0.25
## 11 Locus-C      C      0.25
## 12 Locus-C      D      0.25
## 13 Locus-D      A      0.25
## 14 Locus-D      B      0.25
## 15 Locus-D      C      0.25
## 16 Locus-D      D      0.25</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">adults &lt;-<span class="st"> </span><span class="kw">make_population</span>( freqs, <span class="dt">N=</span><span class="dv">100</span> )
adults$OffID &lt;-<span class="st"> </span><span class="dv">0</span>
adults &lt;-<span class="st"> </span>adults[ , <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">2</span>:<span class="dv">5</span>)]
adults[<span class="dv">1</span>:<span class="dv">5</span>,]</code></pre></div>
<pre><code>##   ID OffID Locus-A Locus-B Locus-C Locus-D
## 1  1     0     A:B     D:D     A:D     C:C
## 2  2     0     A:A     B:D     D:D     B:D
## 3  3     0     A:C     A:D     A:D     C:D
## 4  4     0     C:D     A:C     A:D     C:D
## 5  5     0     A:D     D:D     B:D     A:D</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">offs &lt;-<span class="st"> </span><span class="kw">data.frame</span>()
mom &lt;-<span class="st"> </span>adults[<span class="dv">1</span>,]
for( i in <span class="dv">1</span>:<span class="dv">20</span>){
  dad_id &lt;-<span class="st"> </span><span class="kw">runif</span>( <span class="dv">1</span>, <span class="dt">min=</span><span class="dv">2</span>, <span class="dt">max=</span><span class="dv">100</span>)
  dad &lt;-<span class="st"> </span>adults[dad_id,]
  off &lt;-<span class="st"> </span><span class="kw">mate</span>( mom, dad, <span class="dt">N=</span><span class="dv">1</span> )
  offs &lt;-<span class="st"> </span><span class="kw">rbind</span>( offs, off )
}
offs$OffID &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="dv">20</span>
offs[<span class="dv">1</span>:<span class="dv">5</span>,]</code></pre></div>
<pre><code>##   ID OffID Locus-A Locus-B Locus-C Locus-D
## 1  1     1     A:B     B:D     A:C     A:C
## 2  1     2     B:B     A:D     A:C     B:C
## 3  1     3     A:B     D:D     B:D     C:C
## 4  1     4     A:D     A:D     C:D     A:C
## 5  1     5     A:A     B:D     A:B     C:C</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw">rbind</span>( adults, offs )
data &lt;-<span class="st"> </span>data[ <span class="kw">order</span>(data$ID,data$OffID),]
<span class="kw">rownames</span>(data) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">nrow</span>(data)
data[<span class="dv">1</span>:<span class="dv">10</span>,]</code></pre></div>
<pre><code>##    ID OffID Locus-A Locus-B Locus-C Locus-D
## 1   1     0     A:B     D:D     A:D     C:C
## 2   1     1     A:B     B:D     A:C     A:C
## 3   1     2     B:B     A:D     A:C     B:C
## 4   1     3     A:B     D:D     B:D     C:C
## 5   1     4     A:D     A:D     C:D     A:C
## 6   1     5     A:A     B:D     A:B     C:C
## 7   1     6     A:C     D:D     A:C     B:C
## 8   1     7     A:B     D:D     A:B     B:C
## 9   1     8     A:B     A:D     C:D     C:C
## 10  1     9     A:A     C:D     A:D     B:C</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">f &lt;-<span class="st"> </span><span class="kw">frequencies</span>( data[ data$OffID==<span class="dv">0</span>,] )
excl &lt;-<span class="st"> </span><span class="kw">exclusion_probability</span>( f )
excl</code></pre></div>
<pre><code>##     Locus     Pexcl  PexclMax  Fraction
## 1 Locus-A 0.5036015 0.5039062 0.9993953
## 2 Locus-B 0.5033606 0.5039062 0.9989171
## 3 Locus-C 0.5037248 0.5039062 0.9996400
## 4 Locus-D 0.5037248 0.5039062 0.9996400</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>excl$Pexcl
excl_multilocus &lt;-<span class="st"> </span><span class="dv">1</span> -<span class="st"> </span><span class="kw">prod</span>( <span class="dv">1</span>-p )
excl_multilocus</code></pre></div>
<pre><code>## [1] 0.9392821</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">family &lt;-<span class="st"> </span>data[ data$ID==<span class="dv">1</span>, ]
<span class="kw">minus_mom</span>( family )</code></pre></div>
<pre><code>##    ID OffID Locus-A Locus-B Locus-C Locus-D
## 2   1     1     A:B       B       C       A
## 3   1     2       B       A       C       B
## 4   1     3     A:B       D       B       C
## 5   1     4       D       A       C       A
## 6   1     5       A       B       B       C
## 7   1     6       C       D       C       B
## 8   1     7     A:B       D       B       B
## 9   1     8     A:B       A       C       C
## 10  1     9       A       C     A:D       B
## 11  1    10       A       C       D       C
## 12  1    11       C       D       A       D
## 13  1    12       B       A       D       C
## 14  1    13       D       C       A       D
## 15  1    14       D       C       B       A
## 16  1    15       C       D       B       B
## 17  1    16     A:B       D     A:D       C
## 18  1    17       A       B       A       B
## 19  1    18       C       A     A:D       B
## 20  1    19       D       B       D       D
## 21  1    20       C       B       C       D</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dads &lt;-<span class="st"> </span>adults[<span class="dv">2</span>:<span class="dv">100</span>,]
mom &lt;-<span class="st"> </span>adults[<span class="dv">1</span>,]
off &lt;-<span class="st"> </span>offs[<span class="dv">1</span>,]
for( i in <span class="dv">1</span>:<span class="kw">nrow</span>(dads)){
  dad &lt;-<span class="st"> </span>dads[i,]
  T &lt;-<span class="st"> </span><span class="kw">transition_probability</span>(off,mom,dad)
  if( T &gt;<span class="st"> </span><span class="dv">0</span> )
    <span class="kw">cat</span>(<span class="st">&quot;Father&quot;</span>,i,<span class="st">&quot;may be the real father (T =&quot;</span>,T,<span class="st">&quot;)</span><span class="ch">\n</span><span class="st">&quot;</span>)
}</code></pre></div>
<pre><code>## Father 40 may be the real father (T = 0.0625 )
## Father 48 may be the real father (T = 0.03125 )
## Father 52 may be the real father (T = 0.015625 )
## Father 67 may be the real father (T = 0.03125 )
## Father 68 may be the real father (T = 0.015625 )
## Father 69 may be the real father (T = 0.03125 )
## Father 90 may be the real father (T = 0.015625 )</code></pre>
</div>
<div id="fractional-paternity" class="section level2">
<h2><span class="header-section-number">15.4</span> Fractional Paternity</h2>
<p>In cases where we have more than one putative father, we may want to get an idea of the relative strength of our inferences by comparing the likelihood ratios for all dads.</p>
<ol style="list-style-type: decimal">
<li>We may use arbitrary cut-offs, or<br />
</li>
<li>We may use all non-excluded dads, but weighted by their fractional contributions</li>
</ol>
<p>Conditional Probability</p>
<p><strong>Problem:</strong> We have several putative fathers (<span class="math inline">\(MP_i, MP_j, MP_k, ... , MP_m\)</span>) have been found to have non-zero likelihoods of paternity.</p>
<p><span class="math inline">\(\;\)</span></p>
<p><strong>Question:</strong> What is the relative likelihood of paternity given these putative fathers?</p>
<p>Conditional Probability</p>
<p>Conditional probability determines the likelihood of an event (paternal likelihood) <font class="alert">given</font> that some other event has already happened (not excluded as a potential father).</p>
<p><span class="math display">\[
P(MP=j^*|FP=i,O=k) = \frac{P(O=k|FP=i,MP=j^*)P(MP=j^*|FP=i)}{\sum_{\forall j}P(O=k|FP=i,MP=j^*)P(MP=j^*|FP=i)}
\]</span></p>
<p>If we can assume that <span class="math inline">\(P(MP=j|FP=i) = c\)</span> (e.g., the frequencies of the female and male parents are constant with respect to the individual offspring being considered) then,</p>
<p><span class="math display">\[
P(MP=j^*|FP=i,O=k) = \frac{T(\gamma_k|\alpha_i,\beta_j^*)}{\sum_{\forall k}T(\gamma_k|\alpha_i,\beta_k^*)}
\]</span></p>
<p>Fractional Paternity</p>
<p>Some things to consider when using fractional analyses for paternity.</p>
<ol style="list-style-type: decimal">
<li>Not usually used in human studies.<br />
</li>
<li>Can be considered a <em>prior probability of paternity</em>.<br />
</li>
<li>Can include ecological, spatial, evolutionary components such as differential attractiveness, pollen fertility, output, etc.</li>
<li>Possible tautology</li>
</ol>
<p>Every potential father is assigned paternity, the fraction of <span class="math inline">\(X_{ik}\)</span> on a particular <span class="math inline">\(MP_j\)</span> is proportional to the likelihood ratio.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">frac &lt;-<span class="st"> </span><span class="kw">paternity</span>( offs, mom, dads )
<span class="kw">summary</span>(frac)</code></pre></div>
<pre><code>##      MomID       OffID            DadID             Fij         
##  Min.   :1   Min.   : 1.000   Min.   :  3.00   Min.   :0.04762  
##  1st Qu.:1   1st Qu.: 5.000   1st Qu.: 24.00   1st Qu.:0.09524  
##  Median :1   Median : 9.000   Median : 52.00   Median :0.15385  
##  Mean   :1   Mean   : 9.771   Mean   : 50.39   Mean   :0.18349  
##  3rd Qu.:1   3rd Qu.:15.000   3rd Qu.: 73.00   3rd Qu.:0.22222  
##  Max.   :1   Max.   :20.000   Max.   :100.00   Max.   :0.57143</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">frac[<span class="dv">1</span>:<span class="dv">10</span>,]</code></pre></div>
<pre><code>##    MomID OffID DadID        Fij
## 1      1     1    41 0.30769231
## 2      1     1    49 0.15384615
## 3      1     1    68 0.15384615
## 4      1     1    70 0.15384615
## 5      1     1    53 0.07692308
## 6      1     1    69 0.07692308
## 7      1     1    91 0.07692308
## 8      1     2    75 0.53333333
## 9      1     2    24 0.13333333
## 10     1     2    29 0.06666667</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">t &lt;-<span class="st"> </span><span class="kw">table</span>(frac$OffID)
t</code></pre></div>
<pre><code>## 
##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 
##  7  7  9  4  6  4  8  8  5  3  6  3  3  5  4  7  4  8  3  5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">N_pexcl &lt;-<span class="st"> </span><span class="kw">nrow</span>(dads)*(<span class="dv">1</span>-excl_multilocus)
fit &lt;-<span class="st"> </span><span class="kw">t.test</span>(<span class="kw">as.numeric</span>(t),<span class="dt">mu =</span> N_pexcl)
fit</code></pre></div>
<pre><code>## 
##  One Sample t-test
## 
## data:  as.numeric(t)
## t = -1.2806, df = 19, p-value = 0.2158
## alternative hypothesis: true mean is not equal to 6.011072
## 95 percent confidence interval:
##  4.532946 6.367054
## sample estimates:
## mean of x 
##      5.45</code></pre>
<p>Maternity Analysis</p>
<p>Putative father identified by mother unknown.</p>
<p><span class="math display">\[
\lambda_i = \frac{P(\gamma|\beta)P(\beta)}{P(\gamma)P{\beta})} \\
            = \frac{P(\gamma|\beta)}{P(\gamma)} 
\]</span></p>
<p>where <span class="math inline">\(P(\gamma)\)</span> is the frequency of the offspring genotype in the population. All other things are the same.</p>
<p>Cryptic Gene Flow</p>
<p>Consider the case where:</p>
<ol style="list-style-type: decimal">
<li>You have identified a set of offspring collected from mothers.</li>
<li>Identified a set of fathers that are probabilistically sires of the offspring.</li>
</ol>
</div>
<div id="dispersal-kernels" class="section level2">
<h2><span class="header-section-number">15.5</span> Dispersal Kernels</h2>
<p>Estimating the Disperal Distribution</p>
<p>Once a collection of paternity estimates have been determined, you can use them to estimate a <em>dispersal kernel</em>, describing the probability of paternity as a function of distance from the maternal individual.</p>
<p><img src="amg_files/figure-html/unnamed-chunk-303-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Dispersal Kernels | Distributions</p>
<p>The <font class="alert">form</font> of the distribution is critical for estimation. It determines:</p>
<blockquote>
<ol style="list-style-type: decimal">
<li>The shape of the distribution<br />
</li>
<li>The variance of the distribution</li>
<li>Quantitative estimates and hypotheses you get from the data</li>
</ol>
</blockquote>
<p>Example Kernel Distribution Families</p>
<p><em>Normal Family</em></p>
<p><span class="math display">\[
p(a|x,y) = \frac{1}{\pi a^2}exp\left[-\left( \frac{r}{a} \right)^2\right]
\]</span></p>
<p>where <span class="math inline">\(r = \sqrt{ x^2 + y^2}\)</span> and <span class="math inline">\(a = \sigma \sqrt{2}\)</span>.</p>
<p><span class="math inline">\(\;\)</span></p>
<p>This produces a <em>thin tailed</em> distribution.</p>
<p>Example Kernel Distribution Families</p>
<p><em>Exponential Family</em></p>
<p><span class="math display">\[
p(a,b|x,y) = \frac{b}{2\pi a^2 \Gamma(2/b)}exp\left[ -\left( \frac{r}{a} \right)^b \right]
\]</span></p>
<p>where <span class="math inline">\(\Gamma(a,b)\)</span> is the gamma function and <span class="math inline">\(b\)</span> is a ‘shape’ parameter.</p>
<ol style="list-style-type: decimal">
<li>When <span class="math inline">\(b=1\)</span> This is the exponential distribution.<br />
</li>
<li>When <span class="math inline">\(b=2\)</span> this is the normal function.<br />
</li>
<li>When <span class="math inline">\(b&lt;1\)</span> this is a fat-tailed distribution.</li>
</ol>
<p>Example Kernel Distribution Families</p>
<p>Other distributions you may run across include:</p>
<ol style="list-style-type: decimal">
<li>The Geometric distribution,<br />
</li>
<li>The Weibull family of distributions,<br />
</li>
<li>The 2Dt family of distributions.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">r1 &lt;-<span class="st"> </span><span class="kw">abs</span>(<span class="kw">rnorm</span>(<span class="dv">10000</span>))
r2 &lt;-<span class="st"> </span><span class="kw">rexp</span>(<span class="dv">10000</span>)
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>( <span class="dt">Distribution=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;Normal&quot;</span>,<span class="st">&quot;Exponential&quot;</span>),<span class="dt">each=</span><span class="dv">10000</span>)), <span class="dt">Value=</span><span class="kw">c</span>(r1,r2))
<span class="kw">library</span>(ggplot2)
<span class="kw">ggplot</span>(df,<span class="kw">aes</span>(<span class="dt">x=</span>Value,<span class="dt">fill=</span>Distribution)) +<span class="st"> </span><span class="kw">geom_density</span>(<span class="dt">alpha=</span><span class="fl">0.75</span>) +<span class="st"> </span><span class="kw">theme_bw</span>() +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Frequency&quot;</span>)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-304-1.png" width="672" /></p>
<p>Concerns with kernel estimation</p>
<p>The following are some assumptions that are inherent in the use of dispersal kernels for estimating connectivity.</p>
<ol style="list-style-type: decimal">
<li>All functions are continuous,</li>
<li>All functions assume isotropy in dispersal,<br />
</li>
<li>All functions explicitly assume homogeneity of the dispersal matrix.</li>
</ol>
<p>Skills</p>
<p>In this lecture we covered some rather simple parent/offspring relationships and how we can analyze them. Specifically, you should be comfortable with:</p>
<ol style="list-style-type: decimal">
<li>Understanding the qualities of loci that make for more powerful parentage analyses.<br />
</li>
<li>Be able to estimate single and multilocus exclusion probabilities and understand what they mean.<br />
</li>
<li>Estimate likelihood ratios for paternity given Mother, Offspring, and Putative Male Parent.<br />
</li>
<li>Use fractional paternity and understand conditional probability and how it applies to parentage.</li>
<li>Understand dispersal kernel estimation.</li>
</ol>

</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="pedigrees.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mapping-populations.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"],
"instapaper": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
